// vue3-openlayers/dist/esm/animations/index.js
import { _ as _sfc_main } from "../OlAnimationDrop.vue_vue_type_script_setup_true_lang-gu6vctrg.js";
import { _ as _sfc_main$1 } from "../OlAnimationFade.vue_vue_type_script_setup_true_lang-BP3STVQA.js";
import { _ as _sfc_main$2 } from "../OlAnimationPath.vue_vue_type_script_setup_true_lang-NW95E632.js";
import { _ as _sfc_main$3 } from "../OlAnimationShake.vue_vue_type_script_setup_true_lang-CN2M1HaH.js";
import { _ as _sfc_main$4 } from "../OlAnimationSlide.vue_vue_type_script_setup_true_lang-CGcIiBzJ.js";
import { _ as _sfc_main$5 } from "../OlAnimationTeleport.vue_vue_type_script_setup_true_lang-BC02ZNFx.js";
import { _ as _sfc_main$6 } from "../OlAnimationZoom.vue_vue_type_script_setup_true_lang-DUJRksCK.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlAnimationDrop", _sfc_main);
  registerWithAliases(app, "OlAnimationFade", _sfc_main$1);
  registerWithAliases(app, "OlAnimationPath", _sfc_main$2);
  registerWithAliases(app, "OlAnimationShake", _sfc_main$3);
  registerWithAliases(app, "OlAnimationSlide", _sfc_main$4);
  registerWithAliases(app, "OlAnimationTeleport", _sfc_main$5);
  registerWithAliases(app, "OlAnimationZoom", _sfc_main$6);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlAnimationDrop,
  _sfc_main$1 as OlAnimationFade,
  _sfc_main$2 as OlAnimationPath,
  _sfc_main$3 as OlAnimationShake,
  _sfc_main$4 as OlAnimationSlide,
  _sfc_main$5 as OlAnimationTeleport,
  _sfc_main$6 as OlAnimationZoom,
  install as default,
  install
};
