// vue3-openlayers/dist/esm/sources/index.js
import { _ as _sfc_main } from "../OlSourceBingMaps.vue_vue_type_script_setup_true_lang-CrG8MRAP.js";
import { _ as _sfc_main$1 } from "../OlSourceCluster.vue_vue_type_script_setup_true_lang-aTyQ0aCc.js";
import { _ as _sfc_main$2 } from "../OlSourceImageStatic.vue_vue_type_script_setup_true_lang-BFlKoi4N.js";
import { _ as _sfc_main$3 } from "../OlSourceImageWMS.vue_vue_type_script_setup_true_lang-suS1m91W.js";
import { _ as _sfc_main$4 } from "../OlSourceOSM.vue_vue_type_script_setup_true_lang-C2qMOi5d.js";
import { _ as _sfc_main$5 } from "../OlSourceStadiaMaps.vue_vue_type_script_setup_true_lang-B6KgSOY1.js";
import { _ as _sfc_main$6 } from "../OlSourceTianditu.vue_vue_type_script_setup_true_lang-DFrC_zAx.js";
import { _ as _sfc_main$7 } from "../OlSourceTileArcGISRest.vue_vue_type_script_setup_true_lang-Crl-1Erm.js";
import { _ as _sfc_main$8 } from "../OlSourceTileDebug.vue_vue_type_script_setup_true_lang-C63mdabl.js";
import { _ as _sfc_main$9 } from "../OlSourceGeoTIFF.vue_vue_type_script_setup_true_lang-JDKcZFNU.js";
import { _ as _sfc_main$a } from "../OlSourceTileJSON.vue_vue_type_script_setup_true_lang-Bp4zpTWL.js";
import { _ as _sfc_main$b } from "../OlSourceTileWMS.vue_vue_type_script_setup_true_lang-C5q4j8mH.js";
import { _ as _sfc_main$c } from "../OlSourceVector.vue_vue_type_script_setup_true_lang-D3FZYunU.js";
import { _ as _sfc_main$d } from "../OlSourceVectorTile.vue_vue_type_script_setup_true_lang-FifJi3Ic.js";
import { _ as _sfc_main$e } from "../OlSourceXYZ.vue_vue_type_script_setup_true_lang-BAM2_LNN.js";
import { _ as _sfc_main$f } from "../OlSourceWMTS.vue_vue_type_script_setup_true_lang-DMTysifY.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlSourceBingMaps", _sfc_main);
  registerWithAliases(app, "OlSourceCluster", _sfc_main$1);
  registerWithAliases(app, "OlSourceImageStatic", _sfc_main$2);
  registerWithAliases(app, "OlSourceImageWMS", _sfc_main$3);
  registerWithAliases(app, "OlSourceOSM", _sfc_main$4);
  registerWithAliases(app, "OlSourceStadiaMaps", _sfc_main$5);
  registerWithAliases(app, "OlSourceTianditu", _sfc_main$6);
  registerWithAliases(app, "OlSourceTileArcGISRest", _sfc_main$7);
  registerWithAliases(app, "OlSourceTileDebug", _sfc_main$8);
  registerWithAliases(app, "OlSourceGeoTIFF", _sfc_main$9);
  registerWithAliases(app, "OlSourceTileJSON", _sfc_main$a);
  registerWithAliases(app, "OlSourceTileWMS", _sfc_main$b);
  registerWithAliases(app, "OlSourceVector", _sfc_main$c);
  registerWithAliases(app, "OlSourceVectorTile", _sfc_main$d);
  registerWithAliases(app, "OlSourceXYZ", _sfc_main$e);
  registerWithAliases(app, "OlSourceWMTS", _sfc_main$f);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlSourceBingMaps,
  _sfc_main$1 as OlSourceCluster,
  _sfc_main$9 as OlSourceGeoTIFF,
  _sfc_main$2 as OlSourceImageStatic,
  _sfc_main$3 as OlSourceImageWMS,
  _sfc_main$4 as OlSourceOSM,
  _sfc_main$5 as OlSourceStadiaMaps,
  _sfc_main$6 as OlSourceTianditu,
  _sfc_main$7 as OlSourceTileArcGISRest,
  _sfc_main$8 as OlSourceTileDebug,
  _sfc_main$a as OlSourceTileJSON,
  _sfc_main$b as OlSourceTileWMS,
  _sfc_main$c as OlSourceVector,
  _sfc_main$d as OlSourceVectorTile,
  _sfc_main$f as OlSourceWMTS,
  _sfc_main$e as OlSourceXYZ,
  install as default,
  install
};
