// vue3-openlayers/dist/cjs/animations/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const OlAnimationDrop_vue_vue_type_script_setup_true_lang = require("../OlAnimationDrop.vue_vue_type_script_setup_true_lang-DkoFzeR4.js");
const OlAnimationFade_vue_vue_type_script_setup_true_lang = require("../OlAnimationFade.vue_vue_type_script_setup_true_lang-C3q6J_j2.js");
const OlAnimationPath_vue_vue_type_script_setup_true_lang = require("../OlAnimationPath.vue_vue_type_script_setup_true_lang-BMrmGjv7.js");
const OlAnimationShake_vue_vue_type_script_setup_true_lang = require("../OlAnimationShake.vue_vue_type_script_setup_true_lang-bP6k02Sx.js");
const OlAnimationSlide_vue_vue_type_script_setup_true_lang = require("../OlAnimationSlide.vue_vue_type_script_setup_true_lang-kvig8MyF.js");
const OlAnimationTeleport_vue_vue_type_script_setup_true_lang = require("../OlAnimationTeleport.vue_vue_type_script_setup_true_lang-BdCoj637.js");
const OlAnimationZoom_vue_vue_type_script_setup_true_lang = require("../OlAnimationZoom.vue_vue_type_script_setup_true_lang-DmpNzHQi.js");
const registerWithAliases = require("../registerWithAliases-CfHnojZU.js");
function install(app, options) {
  registerWithAliases.registerWithAliases(app, "OlAnimationDrop", OlAnimationDrop_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlAnimationFade", OlAnimationFade_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlAnimationPath", OlAnimationPath_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlAnimationShake", OlAnimationShake_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlAnimationSlide", OlAnimationSlide_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlAnimationTeleport", OlAnimationTeleport_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlAnimationZoom", OlAnimationZoom_vue_vue_type_script_setup_true_lang._sfc_main);
  if (options) {
    app.provide("ol-options", options);
  }
}
exports.OlAnimationDrop = OlAnimationDrop_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlAnimationFade = OlAnimationFade_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlAnimationPath = OlAnimationPath_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlAnimationShake = OlAnimationShake_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlAnimationSlide = OlAnimationSlide_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlAnimationTeleport = OlAnimationTeleport_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlAnimationZoom = OlAnimationZoom_vue_vue_type_script_setup_true_lang._sfc_main;
exports.default = install;
exports.install = install;
