// vue3-openlayers/dist/esm/interactions/index.js
import { _ as _sfc_main } from "../OlInteractionClusterSelect.vue_vue_type_script_setup_true_lang-Dn6BSaAb.js";
import { _ as _sfc_main$1 } from "../OlInteractionDragBox.vue_vue_type_script_setup_true_lang-dsysIbD2.js";
import { _ as _sfc_main$2 } from "../OlInteractionDragRotate.vue_vue_type_script_setup_true_lang-Bb33EV8G.js";
import { _ as _sfc_main$3 } from "../OlInteractionDragRotateAndZoom.vue_vue_type_script_setup_true_lang-CyS4A4OX.js";
import { _ as _sfc_main$4 } from "../OlInteractionLink.vue_vue_type_script_setup_true_lang-1XNVsA3z.js";
import { _ as _sfc_main$5 } from "../OlInteractionSelect.vue_vue_type_script_setup_true_lang-C-01Dv-D.js";
import { _ as _sfc_main$6 } from "../OlInteractionDraw.vue_vue_type_script_setup_true_lang-aIoGlt0E.js";
import { _ as _sfc_main$7 } from "../OlInteractionModify.vue_vue_type_script_setup_true_lang-C_56vOOP.js";
import { _ as _sfc_main$8 } from "../OlInteractionPointer.vue_vue_type_script_setup_true_lang-JYIAiYcG.js";
import { _ as _sfc_main$9 } from "../OlInteractionSnap.vue_vue_type_script_setup_true_lang-Bo8RDtJV.js";
import { _ as _sfc_main$a } from "../OlInteractionTransform.vue_vue_type_script_setup_true_lang-BRhs_rGw.js";
import { _ as _sfc_main$b } from "../OlInteractionMouseWheelZoom.vue_vue_type_script_setup_true_lang-BR8Ylev-.js";
import { _ as _sfc_main$c } from "../OlInteractionSynchronize.vue_vue_type_script_setup_true_lang-Bf7Xj1Mk.js";
function install(app, options) {
  app.component("OlInteractionClusterSelect", _sfc_main);
  app.component("OlInteractionDragBox", _sfc_main$1);
  app.component("OlInteractionDragRotate", _sfc_main$2);
  app.component(
    "OlInteractionDragRotateAndZoom",
    _sfc_main$3
  );
  app.component("OlInteractionLink", _sfc_main$4);
  app.component("OlInteractionSelect", _sfc_main$5);
  app.component("OlInteractionDraw", _sfc_main$6);
  app.component("OlInteractionModify", _sfc_main$7);
  app.component("OlInteractionPointer", _sfc_main$8);
  app.component("OlInteractionSnap", _sfc_main$9);
  app.component("OlInteractionTransform", _sfc_main$a);
  app.component("OlInteractionMouseWheelZoom", _sfc_main$b);
  app.component("OlInteractionSynchronize", _sfc_main$c);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlInteractionClusterSelect,
  _sfc_main$1 as OlInteractionDragBox,
  _sfc_main$2 as OlInteractionDragRotate,
  _sfc_main$3 as OlInteractionDragRotateAndZoom,
  _sfc_main$6 as OlInteractionDraw,
  _sfc_main$4 as OlInteractionLink,
  _sfc_main$7 as OlInteractionModify,
  _sfc_main$b as OlInteractionMouseWheelZoom,
  _sfc_main$8 as OlInteractionPointer,
  _sfc_main$5 as OlInteractionSelect,
  _sfc_main$9 as OlInteractionSnap,
  _sfc_main$c as OlInteractionSynchronize,
  _sfc_main$a as OlInteractionTransform,
  install as default,
  install
};
