// vue3-openlayers/dist/vue3-openlayers.cjs.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const index = require("./index-D-hPehid.cjs");
require("ol-contextmenu/ol-contextmenu.css");
require("ol-ext/dist/ol-ext.css");
require("ol/ol.css");
exports.Animations = index.index$7;
exports.COMMON_EVENTS = index.COMMON_EVENTS;
exports.Geometries = index.index$4;
exports.IMAGE_SOURCE_EVENTS = index.IMAGE_SOURCE_EVENTS;
exports.Interactions = index.index$6;
exports.LAYER_EVENTS = index.LAYER_EVENTS;
exports.Layers = index.index$1;
exports.Map = index.index;
exports.MapControls = index.index$3;
exports.Providers = index.plugin$1;
exports.Sources = index.index$2;
exports.Styles = index.index$5;
exports.TILE_SOURCE_EVENTS = index.TILE_SOURCE_EVENTS;
exports.VECTOR_SOURCE_EVENTS = index.VECTOR_SOURCE_EVENTS;
exports.default = index.plugin;
exports.install = index.install;
exports.useAnimation = index.useAnimation;
exports.useControl = index.useControl;
exports.useGeometry = index.useGeometry;
exports.useLayer = index.useLayer;
exports.useOpenLayersEvents = index.useOpenLayersEvents;
exports.usePropsAsObjectProperties = index.usePropsAsObjectProperties;
exports.useSource = index.useSource;
