// vue3-openlayers/dist/esm/map/index.js
import { _ as _sfc_main } from "../OlFeature.vue_vue_type_script_setup_true_lang-BQ_Uo_A7.js";
import { _ as _sfc_main$1 } from "../OlGeoLocation.vue_vue_type_script_setup_true_lang-D4rasQBc.js";
import { _ as _sfc_main$2 } from "../OlMap.vue_vue_type_script_setup_true_lang-CDNuAZLN.js";
import { _ as _sfc_main$3 } from "../OlOverlay.vue_vue_type_script_setup_true_lang-C010d8t1.js";
import { _ as _sfc_main$4 } from "../OlProjectionRegister.vue_vue_type_script_setup_true_lang-Bmk_-azE.js";
import { _ as _sfc_main$5 } from "../OlView.vue_vue_type_script_setup_true_lang-BKRD00s5.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlFeature", _sfc_main);
  registerWithAliases(app, "OlGeoLocation", _sfc_main$1);
  registerWithAliases(app, "OlMap", _sfc_main$2);
  registerWithAliases(app, "OlOverlay", _sfc_main$3);
  registerWithAliases(app, "OlProjectionRegister", _sfc_main$4);
  registerWithAliases(app, "OlView", _sfc_main$5);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlFeature,
  _sfc_main$1 as OlGeoLocation,
  _sfc_main$2 as OlMap,
  _sfc_main$3 as OlOverlay,
  _sfc_main$4 as OlProjectionRegister,
  _sfc_main$5 as OlView,
  install as default,
  install
};
