// vue3-openlayers/dist/cjs/composables/index.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const composables_useAnimation = require("./useAnimation.js");
const composables_useControl = require("./useControl.js");
const composables_useSource = require("./useSource.js");
const composables_useLayer = require("./useLayer.js");
const composables_useGeometry = require("./useGeometry.js");
const composables_usePropsAsObjectProperties = require("./usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./useOpenLayersEvents.js");
exports.useAnimation = composables_useAnimation.default;
exports.useControl = composables_useControl.default;
exports.useSource = composables_useSource.default;
exports.useLayer = composables_useLayer.default;
exports.useGeometry = composables_useGeometry.default;
exports.usePropsAsObjectProperties = composables_usePropsAsObjectProperties.default;
exports.COMMON_EVENTS = composables_useOpenLayersEvents.COMMON_EVENTS;
exports.IMAGE_SOURCE_EVENTS = composables_useOpenLayersEvents.IMAGE_SOURCE_EVENTS;
exports.LAYER_EVENTS = composables_useOpenLayersEvents.LAYER_EVENTS;
exports.TILE_SOURCE_EVENTS = composables_useOpenLayersEvents.TILE_SOURCE_EVENTS;
exports.VECTOR_SOURCE_EVENTS = composables_useOpenLayersEvents.VECTOR_SOURCE_EVENTS;
exports.useOpenLayersEvents = composables_useOpenLayersEvents.useOpenLayersEvents;
