// vue3-openlayers/dist/cjs/map/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const OlFeature_vue_vue_type_script_setup_true_lang = require("../OlFeature.vue_vue_type_script_setup_true_lang-BzfLWIGU.js");
const OlGeoLocation_vue_vue_type_script_setup_true_lang = require("../OlGeoLocation.vue_vue_type_script_setup_true_lang-Bj7vrkKX.js");
const OlMap_vue_vue_type_script_setup_true_lang = require("../OlMap.vue_vue_type_script_setup_true_lang-CO-brzHY.js");
const OlOverlay_vue_vue_type_script_setup_true_lang = require("../OlOverlay.vue_vue_type_script_setup_true_lang-C1JIynKY.js");
const OlProjectionRegister_vue_vue_type_script_setup_true_lang = require("../OlProjectionRegister.vue_vue_type_script_setup_true_lang-4ZuL5PPe.js");
const OlView_vue_vue_type_script_setup_true_lang = require("../OlView.vue_vue_type_script_setup_true_lang-Dqydq-ET.js");
const registerWithAliases = require("../registerWithAliases-CfHnojZU.js");
function install(app, options) {
  registerWithAliases.registerWithAliases(app, "OlFeature", OlFeature_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeoLocation", OlGeoLocation_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlMap", OlMap_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlOverlay", OlOverlay_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlProjectionRegister", OlProjectionRegister_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlView", OlView_vue_vue_type_script_setup_true_lang._sfc_main);
  if (options) {
    app.provide("ol-options", options);
  }
}
exports.OlFeature = OlFeature_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeoLocation = OlGeoLocation_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlMap = OlMap_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlOverlay = OlOverlay_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlProjectionRegister = OlProjectionRegister_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlView = OlView_vue_vue_type_script_setup_true_lang._sfc_main;
exports.default = install;
exports.install = install;
