// vue3-openlayers/dist/esm/OlAnimatedClusterLayer.vue_vue_type_script_setup_true_lang-D7IcmE4z.js
import { defineComponent, mergeDefaults, shallowRef, watch, provide, createElementBlock, openBlock, renderSlot } from "vue";
import { Cluster } from "ol/source";
import AnimatedCluster from "ol-ext/layer/AnimatedCluster";
import useLayer from "./composables/useLayer.js";
import { VECTOR_SOURCE_EVENTS } from "./composables/useOpenLayersEvents.js";
import { useDefaults } from "./layers/LayersCommonProps.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlAnimatedClusterLayer",
  props: /* @__PURE__ */ mergeDefaults({
    distance: {},
    animationDuration: {},
    easingFunction: { type: Function },
    animationMethod: { type: Function },
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    renderOrder: { type: Function },
    renderBuffer: {},
    source: {},
    map: {},
    declutter: { type: [String, Number, Boolean] },
    style: { type: [Object, Array, Function, null] },
    background: { type: [String, Function] },
    updateWhileAnimating: { type: Boolean },
    updateWhileInteracting: { type: Boolean },
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "addfeature", "changefeature", "clear", "featuresloadend", "featuresloaderror", "featuresloadstart", "removefeature"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const clusterSource = shallowRef(
      new Cluster({
        distance: props.distance,
        geometryFunction: (feature) => feature.getGeometry()
      })
    );
    const { layer, map } = useLayer(
      AnimatedCluster,
      {
        ...props,
        source: clusterSource.value
      },
      VECTOR_SOURCE_EVENTS
    );
    watch(
      () => props.distance,
      (newValue) => {
        clusterSource.value.setDistance(newValue);
      }
    );
    provide("vectorLayer", clusterSource);
    provide("stylable", layer);
    __expose({
      vectorLayer: layer,
      map
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
