// vue3-openlayers/dist/esm/OlAttributionControl.vue_vue_type_script_setup_true_lang-Ch-Se63f.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import Attribution from "ol/control/Attribution";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlAttributionControl",
  props: {
    className: {},
    target: {},
    collapsible: { type: Boolean },
    collapsed: { type: Boolean, default: true },
    tipLabel: {},
    label: {},
    expandClassName: {},
    collapseLabel: {},
    collapseClassName: {},
    render: {},
    attributions: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(Attribution, properties, attrs);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
