// vue3-openlayers/dist/esm/OlControlBar.vue_vue_type_script_setup_true_lang-Ceux-HyR.js
import { defineComponent, useAttrs, provide, renderSlot } from "vue";
import Bar from "ol-ext/control/Bar";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlControlBar",
  props: {
    className: {},
    group: { type: Boolean, default: true },
    toggleOne: { type: Boolean, default: true },
    autoDeactivate: { type: Boolean },
    controls: {},
    attributes: {},
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(Bar, properties, attrs);
    provide("controlBar", control);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
