// vue3-openlayers/dist/esm/OlContextMenuControl.vue_vue_type_script_setup_true_lang-BPm6R-AQ.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import ContextMenu from "ol-contextmenu";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlContextMenuControl",
  props: {
    width: { default: 150 },
    scrollAt: {},
    eventType: { default: "contextmenu" },
    defaultItems: { type: Boolean, default: true },
    items: { default: () => [] }
  },
  emits: ["change", "error", "propertychange", "beforeopen", "open", "close"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(ContextMenu, properties, attrs, [
      "beforeopen",
      "open",
      "close"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
