// vue3-openlayers/dist/esm/OlFullScreenControl.vue_vue_type_script_setup_true_lang-C_U0swSk.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import FullScreen from "ol/control/FullScreen";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlFullScreenControl",
  props: {
    className: {},
    label: {},
    labelActive: {},
    activeClassName: {},
    inactiveClassName: {},
    tipLabel: {},
    keys: { type: Boolean },
    target: {},
    source: {}
  },
  emits: ["change", "error", "propertychange", "enterfullscreen", "leavefullscreen"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(FullScreen, properties, attrs, [
      "enterfullscreen",
      "leavefullscreen"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
