// vue3-openlayers/dist/esm/OlGeoLocation.vue_vue_type_script_setup_true_lang-D4rasQBc.js
import { defineComponent, computed, ref, renderSlot } from "vue";
import Geolocation from "ol/Geolocation";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlGeoLocation",
  props: {
    tracking: { type: Boolean, default: true },
    trackingOptions: { default: () => ({
      enableHighAccuracy: true
    }) },
    projection: { default: "EPSG:3857" }
  },
  emits: ["change", "error", "propertychange", "change:accuracy", "change:accuracyGeometry", "change:altitude", "change:altitudeAccuracy", "change:heading", "change:position", "change:projection", "change:speed", "change:tracking", "change:trackingOptions"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const properties = usePropsAsObjectProperties(props);
    const geoLoc = computed(() => new Geolocation(properties));
    useOpenLayersEvents(geoLoc, [
      "change:accuracy",
      "change:accuracyGeometry",
      "change:altitude",
      "change:altitudeAccuracy",
      "change:heading",
      "change:position",
      "change:projection",
      "change:speed",
      "change:tracking",
      "change:trackingOptions"
    ]);
    const position = ref([]);
    const accuracy = ref(0);
    const altitude = ref(0);
    const altitudeAccuracy = ref(0);
    const speed = ref(0);
    const heading = ref(0);
    const accuracyGeometry = ref(null);
    __expose({
      geoLoc,
      position,
      accuracy,
      altitude,
      altitudeAccuracy,
      speed,
      heading,
      accuracyGeometry
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default", {
        position: position.value,
        speed: speed.value,
        heading: heading.value,
        accuracy: accuracy.value,
        altitude: altitude.value,
        altitudeAccuracy: altitudeAccuracy.value,
        accuracyGeometry: accuracyGeometry.value
      });
    };
  }
});
export {
  _sfc_main as _
};
