// vue3-openlayers/dist/esm/OlGraticuleLayer.vue_vue_type_script_setup_true_lang-CDq_31Fa.js
import { defineComponent, mergeDefaults, provide, createElementBlock, openBlock, renderSlot } from "vue";
import Graticule from "ol/layer/Graticule";
import useLayer from "./composables/useLayer.js";
import { useDefaults } from "./layers/LayersCommonProps.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlGraticuleLayer",
  props: /* @__PURE__ */ mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    maxLines: {},
    strokeStyle: {},
    targetSize: {},
    showLabels: { type: Boolean },
    lonLabelFormatter: { type: Function },
    latLabelFormatter: { type: Function },
    lonLabelPosition: {},
    latLabelPosition: {},
    lonLabelStyle: {},
    latLabelStyle: {},
    intervals: {},
    wrapX: { type: Boolean },
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = useLayer(Graticule, props);
    provide("vectorLayer", layer);
    provide("stylable", layer);
    __expose({
      vectorLayer: layer
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
