// vue3-openlayers/dist/esm/OlHeatmapLayer.vue_vue_type_script_setup_true_lang-vKx-6K0-.js
import { defineComponent, mergeDefaults, provide, createElementBlock, openBlock, renderSlot } from "vue";
import HeatmapLayer from "ol/layer/Heatmap";
import useLayer from "./composables/useLayer.js";
import { useDefaults } from "./layers/LayersCommonProps.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlHeatmapLayer",
  props: /* @__PURE__ */ mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    source: {},
    map: {},
    render: { type: Function },
    properties: {},
    gradient: {},
    radius: {},
    blur: {},
    weight: { type: [Number, Array, String, Function] },
    filter: { type: [Boolean, Array] },
    variables: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "change:blur", "change:gradient", "change:radius"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = useLayer(HeatmapLayer, props, [
      "change:blur",
      "change:extent",
      "change:gradient",
      "change:radius"
    ]);
    provide("heatmapLayer", layer);
    provide("stylable", layer);
    __expose({
      heatmapLayer: layer
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
