// vue3-openlayers/dist/esm/OlInteractionTransform.vue_vue_type_script_setup_true_lang-BRhs_rGw.js
import { defineComponent, inject, shallowRef, watch, onMounted, onUnmounted, provide, renderSlot } from "vue";
import Transform from "ol-ext/interaction/Transform";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
import { always } from "ol/events/condition";
import { TRUE } from "ol/functions";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionTransform",
  props: {
    filter: {},
    layers: {},
    features: {},
    addCondition: {},
    hitTolerance: {},
    translateFeature: { type: Boolean, default: true },
    translate: { type: Boolean, default: true },
    translateBBox: { type: Boolean },
    stretch: { type: Boolean, default: true },
    scale: { type: Boolean, default: true },
    rotate: { type: Boolean, default: true },
    noFlip: { type: Boolean },
    selection: { type: Boolean, default: TRUE },
    keepAspectRatio: { type: Function, default: always },
    modifyCenter: {},
    enableRotatedTransform: { type: Boolean },
    keepRectangle: { type: Boolean },
    style: {},
    pointRadius: {}
  },
  emits: ["change", "error", "propertychange", "select", "rotatestart", "rotating", "rotateend", "translatestart", "translating", "translateend", "scalestart", "scaling", "scaleend"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    function createTransform() {
      return new Transform(properties);
    }
    const transform = shallowRef(createTransform());
    const { updateOpenLayersEventHandlers } = useOpenLayersEvents(transform, [
      "select",
      "rotatestart",
      "rotating",
      "rotateend",
      "translatestart",
      "translating",
      "translateend",
      "scalestart",
      "scaling",
      "scaleend"
    ]);
    watch(transform, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      transform.value = createTransform();
      updateOpenLayersEventHandlers();
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(transform.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(transform.value);
    });
    provide("stylable", transform);
    __expose({
      transform: transform.value
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
