// vue3-openlayers/dist/esm/OlInteractionMouseWheelZoom.vue_vue_type_script_setup_true_lang-BR8Ylev-.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, renderSlot } from "vue";
import MouseWheelZoom from "ol/interaction/MouseWheelZoom";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlInteractionMouseWheelZoom",
  props: {
    condition: {},
    onFocusOnly: { type: Boolean },
    maxDelta: {},
    duration: {},
    timeout: {},
    useAnchor: { type: Boolean, default: true },
    constrainResolution: { type: Boolean }
  },
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const mouseWheelZoom = computed(() => new MouseWheelZoom(properties));
    watch(mouseWheelZoom, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.getInteractions().forEach((interaction) => {
        if (interaction instanceof MouseWheelZoom) {
          map?.removeInteraction(interaction);
        }
      });
      map?.addInteraction(mouseWheelZoom.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(mouseWheelZoom.value);
    });
    __expose({
      mouseWheelZoom
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
