// vue3-openlayers/dist/esm/OlInteractionPointer.vue_vue_type_script_setup_true_lang-JYIAiYcG.js
import { defineComponent, inject, shallowRef, watch, onMounted, onUnmounted, renderSlot } from "vue";
import Pointer from "ol/interaction/Pointer";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionPointer",
  props: {
    handleEvent: { type: Function },
    stopDown: { type: Function }
  },
  emits: ["change", "error", "propertychange", "down", "move", "up", "drag", "change:active"],
  setup(__props, { expose: __expose, emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const pointer = shallowRef(
      new Pointer({
        ...properties,
        handleDownEvent: (e) => {
          emit("down", e);
          return true;
        },
        handleDragEvent: (e) => {
          emit("drag", e);
          return true;
        },
        handleMoveEvent: (e) => {
          emit("move", e);
        },
        handleUpEvent: (e) => {
          emit("up", e);
          return true;
        }
      })
    );
    useOpenLayersEvents(pointer, ["change:active"]);
    watch(pointer, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(pointer.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(pointer.value);
    });
    __expose({
      pointer
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
