// vue3-openlayers/dist/esm/OlInteractionSelect.vue_vue_type_script_setup_true_lang-C-01Dv-D.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, provide, renderSlot } from "vue";
import Select from "ol/interaction/Select";
import Style from "ol/style/Style";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionSelect",
  props: {
    addCondition: { type: Function },
    condition: { type: Function },
    layers: { type: [Array, Function] },
    style: { type: [Object, Array, Function, null] },
    removeCondition: { type: Function },
    toggleCondition: { type: Function },
    multi: { type: Boolean },
    features: {},
    filter: { type: Function },
    hitTolerance: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const select = computed(
      () => new Select({
        style: new Style(),
        ...properties
      })
    );
    useOpenLayersEvents(select, ["select"]);
    watch(select, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(select.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(select.value);
    });
    provide("stylable", select);
    __expose({
      select
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
