// vue3-openlayers/dist/esm/OlInteractionModify.vue_vue_type_script_setup_true_lang-C_56vOOP.js
import { defineComponent, inject, toRefs, watch, onMounted, onUnmounted, provide, ref, renderSlot } from "vue";
import Modify from "ol/interaction/Modify";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionModify",
  props: {
    condition: { type: Function },
    deleteCondition: { type: Function },
    insertVertexCondition: { type: Function },
    pixelTolerance: {},
    style: { type: [Object, Array, Function] },
    source: {},
    hitDetection: { type: Boolean },
    features: {},
    wrapX: { type: Boolean },
    snapToPointer: { type: Boolean }
  },
  emits: ["change", "error", "propertychange", "modifystart", "modifyend"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const vectorSource = inject("vectorSource", null);
    const {
      source,
      features,
      condition,
      deleteCondition,
      insertVertexCondition,
      pixelTolerance,
      wrapX,
      hitDetection,
      snapToPointer
    } = toRefs(props);
    function createModify() {
      if (!(source?.value || features.value)) {
        console.error(
          `[Vue3-OpenLayers Error] OlInteractionModify: Modify interactions needs either a source or features to work.
      Please provide either the props 'source' or 'feature' or use the component with an '<OlSourceVector>' component.`
        );
      }
      return new Modify({
        source: source.value || vectorSource?.value,
        features: features.value,
        condition: condition.value,
        deleteCondition: deleteCondition.value,
        insertVertexCondition: insertVertexCondition.value,
        pixelTolerance: pixelTolerance.value,
        wrapX: wrapX.value,
        hitDetection: hitDetection.value,
        snapToPointer: snapToPointer.value
      });
    }
    let modify = createModify();
    useOpenLayersEvents(modify, ["modifystart", "modifyend"]);
    watch(
      [
        condition,
        deleteCondition,
        insertVertexCondition,
        pixelTolerance,
        wrapX,
        hitDetection,
        snapToPointer
      ],
      () => {
        modify.dispose();
        map?.removeInteraction(modify);
        modify = createModify();
        map?.addInteraction(modify);
        map?.changed();
      }
    );
    onMounted(() => {
      map?.addInteraction(modify);
    });
    onUnmounted(() => {
      map?.removeInteraction(modify);
    });
    provide("stylable", ref(modify));
    __expose({
      modify
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
