// vue3-openlayers/dist/esm/OlInteractionClusterSelect.vue_vue_type_script_setup_true_lang-Dn6BSaAb.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, provide, renderSlot } from "vue";
import SelectCluster from "ol-ext/interaction/SelectCluster";
import Style from "ol/style/Style";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionClusterSelect",
  props: {
    featureStyle: {},
    selectCluster: { type: Boolean, default: true },
    pointRadius: { default: 7 },
    spiral: { type: Boolean, default: true },
    circleMaxObject: {},
    maxObjects: {},
    animate: { type: Boolean, default: true },
    animationDuration: { default: 500 },
    autoClose: { type: Boolean, default: true },
    addCondition: {},
    condition: {},
    layers: {},
    style: {},
    removeCondition: {},
    toggleCondition: {},
    multi: { type: Boolean, default: false },
    features: {},
    filter: {},
    hitTolerance: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const select = computed(
      () => new SelectCluster({
        style: new Style(),
        ...properties
      })
    );
    useOpenLayersEvents(select, ["select"]);
    watch(select, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(select.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(select.value);
    });
    provide("stylable", select);
    __expose({
      select
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
