// vue3-openlayers/dist/esm/OlInteractionDragBox.vue_vue_type_script_setup_true_lang-dsysIbD2.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, renderSlot } from "vue";
import DragBox from "ol/interaction/DragBox";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionDragBox",
  props: {
    className: {},
    condition: { type: Function },
    minArea: {},
    boxEndCondition: { type: Function },
    onBoxEnd: { type: Function }
  },
  emits: ["change", "error", "propertychange", "boxcancel", "boxdrag", "boxend", "boxstart"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const dragbox = computed(() => new DragBox(properties));
    useOpenLayersEvents(dragbox, ["boxcancel", "boxdrag", "boxend", "boxstart"]);
    watch(dragbox, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(dragbox.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(dragbox.value);
    });
    __expose({
      dragbox
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
