// vue3-openlayers/dist/esm/OlInteractionSnap.vue_vue_type_script_setup_true_lang-Bo8RDtJV.js
import { defineComponent, inject, watch, onMounted, onUnmounted, renderSlot } from "vue";
import Snap from "ol/interaction/Snap";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlInteractionSnap",
  props: {
    features: {},
    source: {},
    edge: { type: Boolean, default: true },
    vertex: { type: Boolean, default: true },
    intersection: { type: Boolean },
    pixelTolerance: {},
    segmenters: {}
  },
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const vectorSource = inject("vectorSource");
    const properties = usePropsAsObjectProperties(props);
    const createSnap = () => new Snap({
      ...properties,
      source: props.source || vectorSource?.value
    });
    let snap = createSnap();
    watch(properties, () => {
      map?.removeInteraction(snap);
      snap = createSnap();
      map?.addInteraction(snap);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(snap);
    });
    onUnmounted(() => {
      map?.removeInteraction(snap);
    });
    __expose({
      snap
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
