// vue3-openlayers/dist/esm/OlInteractionLink.vue_vue_type_script_setup_true_lang-1XNVsA3z.js
import { defineComponent, inject, shallowRef, watch, onMounted, onUnmounted, provide, renderSlot } from "vue";
import Link from "ol/interaction/Link";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionLink",
  props: {
    animate: { type: [Boolean, Object], default: true },
    params: { default: () => ["x", "y", "z", "r", "l"] },
    replace: { type: Boolean, default: false },
    prefix: { default: "" }
  },
  emits: ["change", "error", "propertychange", "change:active"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const link = shallowRef(new Link(properties));
    useOpenLayersEvents(link, ["change:active"]);
    watch(link, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    onMounted(() => {
      map?.addInteraction(link.value);
    });
    onUnmounted(() => {
      map?.removeInteraction(link.value);
    });
    provide("stylable", link);
    __expose({
      link
    });
    return (_ctx, _cache) => {
      return renderSlot(_ctx.$slots, "default");
    };
  }
});
export {
  _sfc_main as _
};
