// vue3-openlayers/dist/esm/OlLayerSwitcherImageControl.vue_vue_type_script_setup_true_lang-BZoTFgp8.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import LayerSwitcherImage from "ol-ext/control/LayerSwitcherImage";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlLayerSwitcherImageControl",
  props: {
    displayInLayerSwitcher: {},
    show_progress: { type: Boolean },
    mouseover: { type: Boolean },
    reordering: { type: Boolean, default: true },
    trash: { type: Boolean },
    oninfo: {},
    extent: { type: Boolean },
    onExtent: {},
    drawDelay: {},
    collapsed: { type: Boolean, default: true },
    layerGroup: {},
    noScroll: { type: Boolean },
    onchangeCheck: {},
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange", "drawlist", "layer:visible", "layer:opacity", "toggle", "reorder-start", "reorder-end"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(LayerSwitcherImage, properties, attrs, [
      "drawlist",
      "layer:visible",
      "layer:opacity",
      "toggle",
      "reorder-start",
      "reorder-end"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
