// vue3-openlayers/dist/esm/OlLayerGroup.vue_vue_type_script_setup_true_lang-DDHBKTpO.js
import { defineComponent, inject, shallowRef, watch, onMounted, onUnmounted, provide, createElementBlock, openBlock, renderSlot } from "vue";
import LayerGroup from "ol/layer/Group";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlLayerGroup",
  props: {
    opacity: { default: 1 },
    visible: { type: Boolean, default: true },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    layers: {},
    properties: { default: () => ({}) },
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean, default: true },
    preview: {}
  },
  emits: ["change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "change:layers"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const layerGroup = shallowRef(new LayerGroup(properties));
    useOpenLayersEvents(layerGroup, [
      "change:extent",
      "change:layers",
      "change:maxResolution",
      "change:maxZoom",
      "change:minResolution",
      "change:minZoom",
      "change:opacity",
      "change:visible",
      "change:zIndex"
    ]);
    const parentLayerGroup = inject("layerGroup", null);
    watch(
      () => properties,
      (newValue) => {
        for (const key in newValue) {
          const keyInObj = key;
          if (newValue[keyInObj] !== void 0) {
            layerGroup.value.set(key, newValue[keyInObj]);
          }
        }
      },
      { deep: true }
    );
    onMounted(() => {
      map?.addLayer(layerGroup.value);
      if (parentLayerGroup) {
        const layerCollection = parentLayerGroup.getLayers();
        layerCollection.push(layerGroup.value);
        parentLayerGroup.setLayers(layerCollection);
      }
    });
    onUnmounted(() => {
      map?.removeLayer(layerGroup.value);
    });
    provide("layerGroup", layerGroup.value);
    __expose({
      layerGroup: layerGroup.value
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
