// vue3-openlayers/dist/esm/OlMousePositionControl.vue_vue_type_script_setup_true_lang-CK_BZWpg.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import MousePosition from "ol/control/MousePosition";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlMousePositionControl",
  props: {
    className: {},
    coordinateFormat: {},
    projection: {},
    render: {},
    target: {},
    placeholder: {},
    wrapX: { type: Boolean, default: true }
  },
  emits: ["change", "error", "propertychange", "change:coordinateFormat", "change:projection"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(MousePosition, properties, attrs, [
      "change:coordinateFormat",
      "change:projection"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
