// vue3-openlayers/dist/esm/OlOverlay.vue_vue_type_script_setup_true_lang-C010d8t1.js
import { defineComponent, inject, ref, shallowRef, onMounted, onUnmounted, watch, watchEffect, createElementBlock, openBlock, renderSlot } from "vue";
import Overlay from "ol/Overlay";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlOverlay",
  props: {
    id: {},
    element: {},
    offset: {},
    position: {},
    positioning: {},
    stopEvent: { type: Boolean, default: true },
    insertFirst: { type: Boolean, default: true },
    autoPan: { type: [Boolean, Object] },
    className: {}
  },
  emits: ["change", "error", "propertychange", "change:element", "change:map", "change:offset", "change:position", "change:positioning"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const htmlContent = ref();
    const properties = usePropsAsObjectProperties(props);
    const overlay = shallowRef(new Overlay(properties));
    useOpenLayersEvents(overlay, [
      "change:element",
      "change:map",
      "change:offset",
      "change:position",
      "change:positioning"
    ]);
    const getOffset = () => overlay.value.getOffset();
    const getPosition = () => overlay.value.getPosition();
    const getPositioning = () => overlay.value.getPositioning();
    const panIntoView = (optPanIntoViewOptions) => overlay.value.panIntoView(optPanIntoViewOptions);
    const setElement = (element) => overlay.value.setElement(element);
    const setOffset = (offset) => overlay.value.setOffset(offset);
    const setPosition = (position) => overlay.value.setPosition(position);
    const setPositioning = (positioning) => overlay.value.setPositioning(positioning);
    function removeOverlay(ov) {
      const removed = map?.removeOverlay(ov);
      if (!removed) {
        console.warn("couldn't find matching overlay to remove", overlay.value);
      }
    }
    onMounted(() => {
      map?.addOverlay(overlay.value);
    });
    onUnmounted(() => removeOverlay(overlay.value));
    watch(
      () => properties,
      (newValue) => {
        for (const key in newValue) {
          const keyInObj = key;
          if (newValue[keyInObj] !== void 0) {
            overlay.value.set(key, newValue[keyInObj]);
          }
        }
      },
      { deep: true }
    );
    watchEffect(
      () => {
        setElement(htmlContent.value);
      },
      {
        flush: "post"
      }
    );
    __expose({
      overlay,
      htmlContent,
      getOffset,
      getPosition,
      getPositioning,
      panIntoView,
      setElement,
      setOffset,
      setPosition,
      setPositioning
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        ref_key: "htmlContent",
        ref: htmlContent
      }, [
        renderSlot(_ctx.$slots, "default", { position: _ctx.position })
      ], 512);
    };
  }
});
export {
  _sfc_main as _
};
