// vue3-openlayers/dist/esm/OlOverviewMapControl.vue_vue_type_script_setup_true_lang-DjSUh1ML.js
import { defineComponent, useAttrs, provide, createElementBlock, openBlock, renderSlot } from "vue";
import OverviewMap from "ol/control/OverviewMap";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlOverviewMapControl",
  props: {
    className: {},
    collapsed: { type: Boolean, default: true },
    collapseLabel: {},
    collapsible: { type: Boolean, default: true },
    label: {},
    layers: {},
    render: {},
    rotateWithView: { type: Boolean },
    target: {},
    tipLabel: {},
    view: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(OverviewMap, properties, attrs);
    provide("overviewMap", control);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
