// vue3-openlayers/dist/esm/OlProfileControl.vue_vue_type_script_setup_true_lang-Cz-5J71K.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import Profile from "ol-ext/control/Profile";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlProfileControl",
  props: {
    className: {},
    title: {},
    style: {},
    selectStyle: {},
    info: {},
    width: {},
    height: {},
    units: {},
    feature: {},
    selectable: { type: Boolean },
    zoomable: { type: Boolean },
    numberFormat: {},
    skipFirst: {},
    element: {},
    render: { type: Function },
    target: {}
  },
  emits: ["change", "error", "propertychange", "over", "out", "show", "dragstart", "dragging", "dragend", "dragcancel"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(Profile, properties, attrs);
    useOpenLayersEvents(control, [
      "over",
      "out",
      "show",
      "dragstart",
      "dragging",
      "dragend",
      "dragcancel"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
