// vue3-openlayers/dist/esm/OlSearchControl.vue_vue_type_script_setup_true_lang-90dpSiOb.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import Search from "ol-ext/control/Search";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSearchControl",
  props: {
    className: {},
    title: {},
    label: {},
    reverseTitle: {},
    placeholder: {},
    reverse: { type: Boolean },
    inputLabel: {},
    collapsed: { type: Boolean, default: true },
    noCollapse: { type: Boolean },
    typing: {},
    minLength: {},
    maxItems: {},
    maxHistory: {},
    getTitle: {},
    autocomplete: {},
    onselect: {},
    centerOnSelect: { type: Boolean },
    zoomOnSelect: { type: [Number, Boolean] },
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(Search, properties, attrs);
    useOpenLayersEvents(control, ["select"]);
    __expose({ control });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
