// vue3-openlayers/dist/esm/OlSourceImageWMS.vue_vue_type_script_setup_true_lang-suS1m91W.js
import { defineComponent, inject, createCommentVNode } from "vue";
import ImageWMS from "ol/source/ImageWMS";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceImageWMS",
  props: {
    attributions: {},
    crossOrigin: {},
    hidpi: { type: Boolean, default: true },
    serverType: { default: "mapserver" },
    imageLoadFunction: {},
    interpolate: { type: Boolean, default: true },
    params: {},
    projection: {},
    ratio: { default: 1 },
    resolutions: {},
    url: {},
    layers: {},
    styles: { default: "" },
    time: {}
  },
  emits: ["imageloadend", "imageloaderror", "imageloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("imageLayer");
    const { source } = useSource(ImageWMS, layer, {
      ...props,
      params: {
        ...props.params,
        LAYERS: props.layers,
        STYLES: props.styles,
        TIME: props.time
      }
    });
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
