// vue3-openlayers/dist/esm/OlSourceVector.vue_vue_type_script_setup_true_lang-D3FZYunU.js
import { defineComponent, inject, watch, provide, createElementBlock, openBlock, renderSlot } from "vue";
import VectorSource from "ol/source/Vector";
import "ol/layer/WebGLVector";
import useSource from "./composables/useSource.js";
import { Collection } from "ol";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceVector",
  props: {
    attributions: {},
    features: {},
    format: {},
    loader: {},
    overlaps: { type: Boolean, default: true },
    strategy: {},
    url: {},
    useSpatialIndex: { type: Boolean, default: true },
    wrapX: { type: Boolean, default: true }
  },
  emits: ["addfeature", "changefeature", "clear", "featuresloadend", "featuresloaderror", "featuresloadstart", "removefeature", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const vectorLayer = inject(
      "vectorLayer",
      null
    );
    const heatmapLayer = inject("heatmapLayer", null);
    const webglVectorLayer = inject(
      "webglVectorLayer",
      null
    );
    const layer = heatmapLayer || vectorLayer || webglVectorLayer;
    const { source, updateSource } = useSource(VectorSource, layer, props);
    watch(
      () => props.features,
      (updatedFeatures, oldFeatures) => {
        if (updatedFeatures !== oldFeatures) {
          source.value.clear();
          if (updatedFeatures) {
            const featureArray = updatedFeatures instanceof Collection ? updatedFeatures.getArray() : updatedFeatures;
            source.value.addFeatures(featureArray);
          }
        }
        updateSource();
      },
      { deep: true }
    );
    watch(
      () => props.url,
      (updatedUrl, newUrl) => {
        if (updatedUrl !== newUrl) {
          source.value.clear();
          if (updatedUrl) {
            source.value.setUrl(updatedUrl);
            source.value.refresh();
          }
        }
        updateSource();
      }
    );
    provide("vectorSource", source);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
