// vue3-openlayers/dist/esm/OlSourceTileArcGISRest.vue_vue_type_script_setup_true_lang-Crl-1Erm.js
import { defineComponent, inject, computed, createCommentVNode } from "vue";
import TileArcGISRest from "ol/source/TileArcGISRest";
import { createXYZ } from "ol/tilegrid";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTileArcGISRest",
  props: {
    attributions: {},
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    params: {},
    hidpi: { type: Boolean, default: true },
    tileGrid: {},
    projection: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    urls: {},
    zDirection: {},
    tileSize: { default: () => [256, 256] }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const tileLayer = inject("tileLayer");
    const tileGrid = computed(() => {
      return props.tileGrid || createXYZ({
        tileSize: props.tileSize
      });
    });
    const { source } = useSource(TileArcGISRest, tileLayer, {
      ...props,
      tileGrid: tileGrid.value
    });
    __expose({
      tileLayer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
