// vue3-openlayers/dist/esm/OlSourceTianditu.vue_vue_type_script_setup_true_lang-DFrC_zAx.js
import { defineComponent, inject, createCommentVNode } from "vue";
import Tianditu from "./sources/TiandituClass.js";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTianditu",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean },
    tileGrid: {},
    projection: { default: "EPSG:3857" },
    reprojectionErrorThreshold: {},
    requestEncoding: { default: "KVP" },
    layer: {},
    style: {},
    tileClass: {},
    tilePixelRatio: {},
    format: {},
    version: { default: "1.0.0" },
    matrixSet: {},
    dimensions: { default: () => ({}) },
    url: {},
    tileLoadFunction: { type: Function, default: (imageTile, src) => {
      imageTile.getImage().src = src;
    } },
    urls: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {},
    tileProxy: {},
    layerType: { default: "img" },
    tk: {},
    isLabel: { type: Boolean, default: false },
    maxZoom: { default: 21 }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("tileLayer");
    const { source } = useSource(Tianditu, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
