// vue3-openlayers/dist/esm/OlSourceWMTS.vue_vue_type_script_setup_true_lang-DMTysifY.js
import { defineComponent, inject, computed, createCommentVNode } from "vue";
import WMTSSource from "ol/source/WMTS";
import TileGridWMTS from "ol/tilegrid/WMTS";
import { get } from "ol/proj";
import { getTopLeft, getWidth } from "ol/extent";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceWMTS",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    projection: {},
    reprojectionErrorThreshold: {},
    requestEncoding: {},
    layer: {},
    tileClass: {},
    tilePixelRatio: {},
    format: {},
    version: {},
    matrixSet: {},
    dimensions: {},
    url: {},
    tileLoadFunction: {},
    urls: {},
    wrapX: { type: Boolean },
    transition: {},
    zDirection: {},
    tileGrid: {},
    styles: {},
    tileZoomLevel: { default: 30 },
    tileMatrixPrefix: { default: "" }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const tileLayer = inject("tileLayer");
    const extent = computed(() => {
      return get(props.projection)?.getExtent();
    });
    const origin = computed(() => {
      return extent.value ? getTopLeft(extent.value) : void 0;
    });
    const size = computed(() => {
      return extent.value ? getWidth(extent.value) / 256 : 0;
    });
    const tileGrid = computed(() => {
      if (props.tileGrid) {
        return props.tileGrid;
      }
      const resolutions = [props.tileZoomLevel];
      const matrixIds = [`${props.tileZoomLevel}`];
      for (let z = 0; z < props.tileZoomLevel; ++z) {
        resolutions[z] = size.value / Math.pow(2, z);
        matrixIds[z] = props.tileMatrixPrefix + z;
      }
      return new TileGridWMTS({
        origin: origin.value,
        resolutions,
        matrixIds
      });
    });
    const { source } = useSource(WMTSSource, tileLayer, {
      ...props,
      style: props.styles,
      tileGrid: tileGrid.value
    });
    __expose({
      tileLayer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
