// vue3-openlayers/dist/esm/OlSourceBingMaps.vue_vue_type_script_setup_true_lang-CrG8MRAP.js
import { defineComponent, inject, createCommentVNode } from "vue";
import BingMaps from "ol/source/BingMaps";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceBingMaps",
  props: {
    cacheSize: {},
    hidpi: { type: Boolean },
    culture: {},
    imagerySet: {},
    interpolate: { type: Boolean, default: true },
    maxZoom: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {},
    placeholderTiles: { type: Boolean },
    apiKey: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("tileLayer");
    const { source } = useSource(BingMaps, layer, {
      ...props,
      key: props.apiKey
    });
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
