// vue3-openlayers/dist/esm/OlSourceVectorTile.vue_vue_type_script_setup_true_lang-FifJi3Ic.js
import { defineComponent, inject, provide, createElementBlock, openBlock, renderSlot } from "vue";
import VectorTileSource from "ol/source/VectorTile";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceVectorTile",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    extent: {},
    format: {},
    overlaps: { type: Boolean, default: true },
    projection: {},
    state: {},
    tileClass: {},
    maxZoom: {},
    minZoom: {},
    tileSize: {},
    maxResolution: {},
    tileGrid: {},
    tileLoadFunction: {},
    tileUrlFunction: {},
    url: {},
    transition: {},
    urls: {},
    wrapX: { type: Boolean, default: true },
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const vectorTileLayer = inject(
      "vectorTileLayer",
      null
    );
    const { source } = useSource(VectorTileSource, vectorTileLayer, props);
    provide("vectorSource", source);
    __expose({
      vectorTileLayer,
      source
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
