// vue3-openlayers/dist/esm/OlSourceStadiaMaps.vue_vue_type_script_setup_true_lang-B6KgSOY1.js
import { defineComponent, inject, createCommentVNode } from "vue";
import StadiaMaps from "ol/source/StadiaMaps";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceStadiaMaps",
  props: {
    cacheSize: {},
    interpolate: { type: Boolean, default: true },
    layer: {},
    minZoom: {},
    maxZoom: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    transition: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    zDirection: {},
    apiKey: {},
    retina: { type: Boolean }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("tileLayer");
    const { source } = useSource(StadiaMaps, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
