// vue3-openlayers/dist/esm/OlSourceTileJSON.vue_vue_type_script_setup_true_lang-Bp4zpTWL.js
import { defineComponent, inject, createCommentVNode } from "vue";
import TileJSON from "ol/source/TileJSON";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTileJSON",
  props: {
    attributions: {},
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    jsonp: { type: Boolean },
    reprojectionErrorThreshold: {},
    tileJSON: {},
    tileLoadFunction: {},
    tileSize: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("tileLayer");
    const { source } = useSource(TileJSON, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
