// vue3-openlayers/dist/esm/OlSourceXYZ.vue_vue_type_script_setup_true_lang-BAM2_LNN.js
import { defineComponent, inject, watch, createCommentVNode } from "vue";
import XYZ from "ol/source/XYZ";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceXYZ",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    projection: {},
    reprojectionErrorThreshold: {},
    maxZoom: {},
    minZoom: {},
    maxResolution: {},
    tileGrid: {},
    tileLoadFunction: {},
    tilePixelRatio: {},
    tileSize: {},
    gutter: {},
    tileUrlFunction: {},
    url: {},
    urls: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("tileLayer");
    const { source, updateSource } = useSource(XYZ, layer, props, [], (source2) => {
      if (props.url) {
        source2.setUrl(props.url);
      }
    });
    watch(
      () => props.url,
      () => updateSource()
    );
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
