// vue3-openlayers/dist/esm/OlSourceCluster.vue_vue_type_script_setup_true_lang-aTyQ0aCc.js
import { defineComponent, inject, provide, createElementBlock, openBlock, renderSlot } from "vue";
import Cluster from "ol/source/Cluster";
import useSource from "./composables/useSource.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceCluster",
  props: {
    attributions: {},
    distance: {},
    minDistance: {},
    geometryFunction: {},
    createCluster: {},
    source: {},
    wrapX: { type: Boolean, default: true }
  },
  emits: ["addfeature", "changefeature", "clear", "featuresloadend", "featuresloaderror", "featuresloadstart", "removefeature", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = inject("vectorLayer");
    const { source } = useSource(Cluster, layer, props);
    provide("vectorLayer", source);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
