// vue3-openlayers/dist/esm/OlStyleText.vue_vue_type_script_setup_true_lang-qgBomPf-.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, provide, createElementBlock, openBlock, renderSlot } from "vue";
import Text from "ol/style/Text";
import Fill from "ol/style/Fill";
import Stroke from "ol/style/Stroke";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { Interaction } from "ol/interaction";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlStyleText",
  props: {
    font: {},
    maxAngle: {},
    offsetX: {},
    offsetY: {},
    overflow: { type: Boolean },
    placement: {},
    repeat: {},
    scale: {},
    rotateWithView: { type: Boolean },
    keepUpright: { type: Boolean },
    rotation: {},
    text: {},
    textAlign: {},
    justify: {},
    textBaseline: {},
    fill: {},
    stroke: {},
    backgroundFill: {},
    backgroundStroke: {},
    padding: {},
    declutterMode: {}
  },
  setup(__props) {
    const props = __props;
    const style = inject("style", null);
    const styledObj = inject(
      "styledObj",
      null
    );
    const properties = usePropsAsObjectProperties(props);
    const createText = (properties2) => {
      const innerProperties = properties2;
      const options = {
        ...innerProperties
      };
      if (properties2.fill) {
        options.fill = new Fill({ color: properties2.fill });
      }
      if (properties2.stroke) {
        options.stroke = new Stroke(properties2.stroke);
      }
      if (properties2.backgroundFill) {
        options.backgroundFill = new Fill({ color: properties2.backgroundFill });
      }
      if (properties2.backgroundStroke) {
        options.backgroundStroke = new Stroke(properties2.backgroundStroke);
      }
      return new Text(options);
    };
    const textContent = computed(() => createText(properties));
    const applyStyle = () => {
      style?.value?.setText(new Text());
      style?.value?.setText(textContent.value);
      if (styledObj?.value instanceof Interaction) {
        styledObj.value.changed();
      }
    };
    watch(properties, () => applyStyle());
    watch(
      () => style?.value,
      () => applyStyle()
    );
    onMounted(() => style?.value?.setText(textContent.value));
    onUnmounted(() => style?.value?.setText(new Text()));
    provide("style", textContent);
    provide("styledObj", styledObj);
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
