// vue3-openlayers/dist/esm/OlStyleStroke.vue_vue_type_script_setup_true_lang-DZedA3zE.js
import { defineComponent, inject, watch, onMounted, onUnmounted, createCommentVNode } from "vue";
import Stroke from "ol/style/Stroke";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { Interaction } from "ol/interaction";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlStyleStroke",
  props: {
    color: {},
    lineCap: {},
    lineJoin: {},
    lineDash: {},
    lineDashOffset: {},
    miterLimit: {},
    width: { default: 1 }
  },
  setup(__props) {
    const props = __props;
    const style = inject("style", null);
    const styledObj = inject(
      "styledObj",
      null
    );
    const circle = inject("circle", null);
    const properties = usePropsAsObjectProperties(props);
    if (style != null && circle == null) {
      let stroke = new Stroke(properties);
      style.value?.setStroke(stroke);
      const applyStroke = () => {
        style?.value?.setStroke(new Stroke());
        stroke = new Stroke(properties);
        style?.value?.setStroke(stroke);
      };
      watch(properties, () => {
        applyStroke();
      });
      watch(style, () => {
        applyStroke();
      });
      onMounted(() => {
        style?.value?.setStroke(stroke);
      });
      onUnmounted(() => {
        style?.value?.setStroke(new Stroke());
      });
    } else if (circle != null) {
      const applyStroketoCircle = (innerProperties) => {
        circle?.value?.getStroke()?.setColor(innerProperties.color || "");
        circle?.value?.getStroke()?.setLineCap(innerProperties.lineCap);
        circle?.value?.getStroke()?.setLineDash(innerProperties.lineDash || []);
        circle?.value?.getStroke()?.setLineDashOffset(innerProperties.lineDashOffset);
        circle?.value?.getStroke()?.setLineJoin(innerProperties.lineJoin);
        circle?.value?.getStroke()?.setMiterLimit(innerProperties.miterLimit);
        circle?.value?.getStroke()?.setWidth(innerProperties.width);
        circle?.value?.setRadius(circle?.value?.getRadius());
        if (styledObj?.value instanceof Interaction) {
          styledObj.value.changed();
        }
      };
      applyStroketoCircle(properties);
      watch(properties, (newVal) => {
        applyStroketoCircle(newVal);
      });
      watch(circle, () => {
        applyStroketoCircle(properties);
      });
    }
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
