// vue3-openlayers/dist/esm/OlStyleFlowLine.vue_vue_type_script_setup_true_lang-DBMVv7NG.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, createElementBlock, openBlock, renderSlot } from "vue";
import FlowLine from "ol-ext/style/FlowLine";
import Draw from "ol/interaction/Draw";
import Modify from "ol/interaction/Modify";
import Feature from "ol/Feature";
import Style from "ol/style/Style";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import "ol/interaction";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlStyleFlowLine",
  props: {
    visible: { type: Boolean },
    width: {},
    width2: {},
    color: {},
    color2: {},
    arrowClor: {},
    lineCap: { default: "butt" },
    arrowSize: {},
    arrow: {},
    noOverlap: { type: Boolean },
    offset0: {},
    offset1: {},
    overrideStyleFunction: {}
  },
  setup(__props, { expose: __expose }) {
    const props = __props;
    const styledObj = inject(
      "stylable",
      null
    );
    const properties = usePropsAsObjectProperties(props);
    const style = computed(() => new FlowLine(properties));
    const setStyle = (val) => {
      if (!styledObj?.value) {
        return;
      }
      if (styledObj.value instanceof Draw || styledObj?.value instanceof Modify) {
        styledObj.value.getOverlay().setStyle(val);
        styledObj.value.dispatchEvent("styleChanged");
        return;
      }
      try {
        if (styledObj.value instanceof Feature) {
          styledObj.value.setStyle(val);
        }
        styledObj.value.changed();
        styledObj.value.dispatchEvent("styleChanged");
      } catch (error) {
        if (styledObj?.value) {
          styledObj.value.style_ = val;
          styledObj.value.values_.style = val;
          styledObj.value.changed();
          styledObj.value.dispatchEvent("styleChanged");
        }
      }
    };
    const styleFunc = computed(() => {
      return (feature, resolution) => {
        if (properties.overrideStyleFunction) {
          properties.overrideStyleFunction(feature, style.value, resolution);
        }
        return style.value;
      };
    });
    watch(
      properties,
      () => setStyle(properties.overrideStyleFunction ? styleFunc.value : style.value)
    );
    onMounted(
      () => setStyle(properties.overrideStyleFunction ? styleFunc.value : style.value)
    );
    onUnmounted(() => setStyle(new Style()));
    __expose({
      style
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
