// vue3-openlayers/dist/esm/OlStyleCircle.vue_vue_type_script_setup_true_lang-J9mmiSKo.js
import { defineComponent, inject, computed, watch, onMounted, onUnmounted, provide, createElementBlock, openBlock, renderSlot } from "vue";
import CircleStyle from "ol/style/Circle";
import Fill from "ol/style/Fill";
import Stroke from "ol/style/Stroke";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import { Interaction } from "ol/interaction";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlStyleCircle",
  props: {
    radius: {},
    scale: {}
  },
  setup(__props) {
    const props = __props;
    const style = inject("style", null);
    const styledObj = inject(
      "styledObj",
      null
    );
    const properties = usePropsAsObjectProperties(props);
    const createCircleStyle = (innerProperties) => {
      return new CircleStyle({
        ...innerProperties,
        fill: new Fill(),
        stroke: new Stroke()
      });
    };
    const circle = computed(() => createCircleStyle(properties));
    const applyStyle = () => {
      style?.value?.setImage(circle.value);
      if (styledObj?.value instanceof Interaction) {
        styledObj.value.changed();
      }
    };
    watch(properties, () => applyStyle());
    watch(
      () => style,
      () => applyStyle()
    );
    onMounted(() => applyStyle());
    onUnmounted(() => {
      style?.value?.setImage(null);
    });
    provide("circle", circle);
    provide("styledObj", styledObj);
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
