// vue3-openlayers/dist/esm/OlToggleControl.vue_vue_type_script_setup_true_lang-BctgQeYP.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import Toggle from "ol-ext/control/Toggle";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlToggleControl",
  props: {
    className: {},
    classButton: {},
    title: {},
    html: {},
    interaction: {},
    active: { type: Boolean },
    disable: { type: Boolean },
    bar: {},
    autoActive: { type: Boolean },
    onToggle: { type: Function },
    attributes: {},
    element: {},
    render: { type: Function },
    target: {}
  },
  emits: ["change", "error", "propertychange", "change:active", "change:disable"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(Toggle, properties, attrs, [
      "change:active",
      "change:disable"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
