// vue3-openlayers/dist/esm/OlVectorImageLayer.vue_vue_type_script_setup_true_lang-DHhHgHh4.js
import { defineComponent, mergeDefaults, provide, createElementBlock, openBlock, renderSlot } from "vue";
import VectorImageLayer from "ol/layer/VectorImage";
import useLayer from "./composables/useLayer.js";
import { useDefaults } from "./layers/LayersCommonProps.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlVectorImageLayer",
  props: /* @__PURE__ */ mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    renderOrder: { type: Function },
    renderBuffer: {},
    source: {},
    map: {},
    declutter: { type: [String, Number, Boolean] },
    style: { type: [Object, Array, Function, null] },
    background: { type: [String, Function] },
    imageRatio: {},
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = useLayer(VectorImageLayer, props);
    provide("vectorLayer", layer);
    provide("stylable", layer);
    __expose({
      vectorImageLayer: layer
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
