// vue3-openlayers/dist/esm/OlView.vue_vue_type_script_setup_true_lang-BKRD00s5.js
import { defineComponent, inject, onMounted, watch, createCommentVNode } from "vue";
import "ol/proj/Projection";
import View from "ol/View";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
import projectionFromProperties from "./helpers/projection.js";
import { useOpenLayersEvents } from "./composables/useOpenLayersEvents.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlView",
  props: {
    center: {},
    constrainRotation: { type: [Number, Boolean] },
    enableRotation: { type: Boolean, default: true },
    extent: {},
    constrainOnlyCenter: { type: Boolean },
    smoothExtentConstraint: { type: Boolean, default: true },
    maxResolution: {},
    minResolution: {},
    maxZoom: {},
    minZoom: {},
    multiWorld: { type: Boolean },
    constrainResolution: { type: Boolean },
    smoothResolutionConstraint: { type: Boolean, default: true },
    showFullExtent: { type: Boolean },
    projection: {},
    resolution: {},
    resolutions: {},
    rotation: {},
    zoom: {},
    zoomFactor: {},
    padding: {}
  },
  emits: ["change", "error", "propertychange", "change:center", "change:resolution", "change:rotation"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = inject("map");
    const properties = usePropsAsObjectProperties(props);
    const createProp = () => ({
      ...properties,
      projection: projectionFromProperties(properties.projection)
    });
    const view = new View(createProp());
    useOpenLayersEvents(view, [
      "change:center",
      "change:resolution",
      "change:rotation"
    ]);
    onMounted(() => {
      map?.setView(view);
    });
    watch(properties, () => {
      const pr = createProp();
      view.setProperties(pr);
      view.applyOptions_(pr);
    });
    const adjustCenter = (deltaCoordinates) => view.adjustCenter(deltaCoordinates);
    const adjustResolution = (ratio, optAnchor) => view.adjustResolution(ratio, optAnchor);
    const adjustRotation = (delta, optAnchor) => view.adjustRotation(delta, optAnchor);
    const adjustZoom = (delta, optAnchor) => view.adjustZoom(delta, optAnchor);
    const animate = (...args) => view.animate(...args);
    const beginInteraction = () => view.beginInteraction();
    const calculateExtent = (optSize) => view.calculateExtent(optSize);
    const cancelAnimations = () => view.cancelAnimations();
    const centerOn = (coordinate, size, position) => view.centerOn(coordinate, size, position);
    const changed = () => view.changed();
    const dispatchEvent = (event) => view.dispatchEvent(event);
    const endInteraction = (optDuration, optResolutionDirection, optAnchor) => view.endInteraction(optDuration, optResolutionDirection, optAnchor);
    const fit = (geometryOrExtent, options) => view.fit(geometryOrExtent, options);
    const get = (key) => view.get(key);
    const getAnimating = () => view.getAnimating();
    const getCenter = () => view.getCenter();
    const getInteracting = () => view.getInteracting();
    const getKeys = () => view.getKeys();
    const getMaxResolution = () => view.getMaxResolution();
    const getMaxZoom = () => view.getMaxZoom();
    const getMinResolution = () => view.getMinResolution();
    const getMinZoom = () => view.getMinZoom();
    const getProjection = () => view.getProjection();
    const getProperties = () => view.getProperties();
    const getResolution = () => view.getResolution();
    const getResolutionForExtent = (extent, size) => view.getResolutionForExtent(extent, size);
    const getResolutionForZoom = (zoom) => view.getResolutionForZoom(zoom);
    const getResolutions = () => view.getResolutions();
    const getRevision = () => view.getRevision();
    const getRotation = () => view.getRotation();
    const getZoom = () => view.getZoom();
    const getZoomForResolution = (resolution) => view.getZoomForResolution(resolution);
    const setCenter = (center) => view.setCenter(center);
    const setConstrainResolution = (enabled) => view.setConstrainResolution(enabled);
    const setMaxZoom = (zoom) => view.setMaxZoom(zoom);
    const setMinZoom = (zoom) => view.setMinZoom(zoom);
    const setResolution = (resolution) => view.setResolution(resolution);
    const setRotation = (rotation) => view.setRotation(rotation);
    const setZoom = (zoom) => view.setZoom(zoom);
    __expose({
      view,
      adjustCenter,
      adjustResolution,
      adjustRotation,
      adjustZoom,
      animate,
      beginInteraction,
      calculateExtent,
      cancelAnimations,
      centerOn,
      changed,
      dispatchEvent,
      endInteraction,
      fit,
      get,
      getAnimating,
      getCenter,
      getInteracting,
      getKeys,
      getMaxResolution,
      getMaxZoom,
      getMinResolution,
      getMinZoom,
      getProjection,
      getProperties,
      getResolution,
      getResolutionForExtent,
      getResolutionForZoom,
      getResolutions,
      getRevision,
      getRotation,
      getZoom,
      getZoomForResolution,
      setCenter,
      setConstrainResolution,
      setMaxZoom,
      setMinZoom,
      setResolution,
      setRotation,
      setZoom
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
