// vue3-openlayers/dist/esm/OlVideoRecorderControl.vue_vue_type_script_setup_true_lang-BoAdojYw.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import VideoRecorder from "ol-ext/control/VideoRecorder";
import { F as FileSaver_minExports } from "./FileSaver.min-DDsuRhxc.js";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlVideoRecorderControl",
  props: {
    className: {},
    frameRate: {},
    videoBitsPerSecond: {},
    videoTarget: {},
    element: {},
    render: {},
    target: {},
    downloadName: { default: "mapVideo.mp4" }
  },
  emits: ["change", "error", "propertychange", "start", "stop", "pause", "result"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(VideoRecorder, properties, attrs, [
      "start",
      "stop",
      "pause",
      "resume"
    ]);
    control.value.on(
      // @ts-expect-error because of wrong typings
      "stop",
      (event) => {
        if (props.downloadName) {
          FileSaver_minExports.saveAs(event.videoURL, props.downloadName);
        }
      }
    );
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
