// vue3-openlayers/dist/esm/OlWebglTileLayer.vue_vue_type_script_setup_true_lang-DjEqRJRE.js
import { defineComponent, mergeDefaults, provide, createElementBlock, openBlock, renderSlot } from "vue";
import TileLayer from "ol/layer/WebGLTile";
import useLayer from "./composables/useLayer.js";
import { useDefaults } from "./layers/LayersCommonProps.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlWebglTileLayer",
  props: /* @__PURE__ */ mergeDefaults({
    style: {},
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    preload: {},
    source: {},
    sources: { type: [Array, Function] },
    map: {},
    useInterimTilesOnError: { type: Boolean },
    cacheSize: {},
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "change:useInterimTilesOnError", "change:preload"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = useLayer(TileLayer, props, [
      "change:useInterimTilesOnError",
      "change:preload"
    ]);
    provide("tileLayer", layer);
    __expose({
      tileLayer: layer
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
