// vue3-openlayers/dist/esm/OlWebglVectorLayer.vue_vue_type_script_setup_true_lang-8_3dH3pz.js
import { defineComponent, mergeDefaults, provide, watch, createElementBlock, openBlock, renderSlot } from "vue";
import useLayer from "./composables/useLayer.js";
import { useDefaults } from "./layers/LayersCommonProps.js";
import WebGLVectorLayer from "ol/layer/WebGLVector";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlWebglVectorLayer",
  props: /* @__PURE__ */ mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    source: {},
    variables: {},
    background: { type: [String, Function] },
    disableHitDetection: { type: Boolean },
    properties: {},
    map: {},
    render: { type: Function },
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {},
    styles: {}
  }, useDefaults({
    styles: () => ({
      "shape-points": 1,
      "shape-radius": 10,
      "shape-opacity": 0.5,
      "shape-fill-color": "blue"
    })
  })),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = useLayer(WebGLVectorLayer, props);
    provide("webglVectorLayer", layer);
    watch(
      () => props.variables,
      (newVariables) => {
        if (newVariables) {
          layer.value.updateStyleVariables(newVariables);
        }
      }
    );
    __expose({
      webglVectorLayer: layer
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
export {
  _sfc_main as _
};
