// vue3-openlayers/dist/esm/OlZoneControl.vue_vue_type_script_setup_true_lang-CIKm3j5y.js
import { defineComponent, useAttrs, createCommentVNode } from "vue";
import MapZone from "ol-ext/control/MapZone";
import useControl from "./composables/useControl.js";
import usePropsAsObjectProperties from "./composables/usePropsAsObjectProperties.js";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "OlZoneControl",
  props: {
    className: { default: "ol-mapzone" },
    layer: {},
    projection: { default: "EPSG:3857" },
    centerOnClick: { type: Boolean, default: true },
    element: {},
    render: {},
    target: {},
    zones: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = useAttrs();
    const properties = usePropsAsObjectProperties(props);
    const { control } = useControl(MapZone, properties, attrs, ["select"]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return createCommentVNode("", true);
    };
  }
});
export {
  _sfc_main as _
};
