// vue3-openlayers/dist/esm/layers/index.js
import { layersCommonDefaultProps, useDefaults } from "./LayersCommonProps.js";
import { _ as _sfc_main } from "../OlAnimatedClusterLayer.vue_vue_type_script_setup_true_lang-D7IcmE4z.js";
import { _ as _sfc_main$1 } from "../OlGraticuleLayer.vue_vue_type_script_setup_true_lang-CDq_31Fa.js";
import { _ as _sfc_main$2 } from "../OlHeatmapLayer.vue_vue_type_script_setup_true_lang-vKx-6K0-.js";
import { _ as _sfc_main$3 } from "../OlImageLayer.vue_vue_type_script_setup_true_lang-DsVvvaqT.js";
import { _ as _sfc_main$4 } from "../OlLayerGroup.vue_vue_type_script_setup_true_lang-DDHBKTpO.js";
import { _ as _sfc_main$5 } from "../OlTileLayer.vue_vue_type_script_setup_true_lang-Bt78UPkH.js";
import { _ as _sfc_main$7 } from "../OlVectorLayer.vue_vue_type_script_setup_true_lang-Cfj4aaxQ.js";
import { _ as _sfc_main$8 } from "../OlVectorTileLayer.vue_vue_type_script_setup_true_lang-B3bPdNI0.js";
import { _ as _sfc_main$6 } from "../OlVectorImageLayer.vue_vue_type_script_setup_true_lang-DHhHgHh4.js";
import { _ as _sfc_main$9 } from "../OlWebglTileLayer.vue_vue_type_script_setup_true_lang-DjEqRJRE.js";
import { _ as _sfc_main$a } from "../OlWebglVectorLayer.vue_vue_type_script_setup_true_lang-8_3dH3pz.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlAnimatedClusterLayer", _sfc_main);
  registerWithAliases(app, "OlGraticuleLayer", _sfc_main$1);
  registerWithAliases(app, "OlHeatmapLayer", _sfc_main$2);
  registerWithAliases(app, "OlImageLayer", _sfc_main$3);
  registerWithAliases(app, "OlLayerGroup", _sfc_main$4);
  registerWithAliases(app, "OlTileLayer", _sfc_main$5);
  registerWithAliases(app, "OlVectorImageLayer", _sfc_main$6);
  registerWithAliases(app, "OlVectorLayer", _sfc_main$7);
  registerWithAliases(app, "OlVectorTileLayer", _sfc_main$8);
  registerWithAliases(app, "OlWebglTileLayer", _sfc_main$9);
  registerWithAliases(app, "OlWebglVectorLayer", _sfc_main$a);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlAnimatedClusterLayer,
  _sfc_main$1 as OlGraticuleLayer,
  _sfc_main$2 as OlHeatmapLayer,
  _sfc_main$3 as OlImageLayer,
  _sfc_main$4 as OlLayerGroup,
  _sfc_main$5 as OlTileLayer,
  _sfc_main$6 as OlVectorImageLayer,
  _sfc_main$7 as OlVectorLayer,
  _sfc_main$8 as OlVectorTileLayer,
  _sfc_main$9 as OlWebglTileLayer,
  _sfc_main$a as OlWebglVectorLayer,
  install as default,
  install,
  layersCommonDefaultProps,
  useDefaults as useLayerDefaults
};
