// vue3-openlayers/dist/cjs/geometries/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const OlGeomCircle_vue_vue_type_script_setup_true_lang = require("../OlGeomCircle.vue_vue_type_script_setup_true_lang-CCBJQ3Wl.js");
const OlGeomLineString_vue_vue_type_script_setup_true_lang = require("../OlGeomLineString.vue_vue_type_script_setup_true_lang-D627LXgX.js");
const OlGeomMultiLineString_vue_vue_type_script_setup_true_lang = require("../OlGeomMultiLineString.vue_vue_type_script_setup_true_lang-rpBQA3H2.js");
const OlGeomMultiPoint_vue_vue_type_script_setup_true_lang = require("../OlGeomMultiPoint.vue_vue_type_script_setup_true_lang-OZQg54zJ.js");
const OlGeomMultiPolygon_vue_vue_type_script_setup_true_lang = require("../OlGeomMultiPolygon.vue_vue_type_script_setup_true_lang-CFROciCY.js");
const OlGeomPoint_vue_vue_type_script_setup_true_lang = require("../OlGeomPoint.vue_vue_type_script_setup_true_lang-DBH5Xe3F.js");
const OlGeomPolygon_vue_vue_type_script_setup_true_lang = require("../OlGeomPolygon.vue_vue_type_script_setup_true_lang-BDXz1Mkp.js");
const registerWithAliases = require("../registerWithAliases-CfHnojZU.js");
function install(app, options) {
  registerWithAliases.registerWithAliases(app, "OlGeomCircle", OlGeomCircle_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeomLineString", OlGeomLineString_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeomMultiLineString", OlGeomMultiLineString_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeomMultiPoint", OlGeomMultiPoint_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeomMultiPolygon", OlGeomMultiPolygon_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeomPoint", OlGeomPoint_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGeomPolygon", OlGeomPolygon_vue_vue_type_script_setup_true_lang._sfc_main);
  if (options) {
    app.provide("ol-options", options);
  }
}
exports.OlGeomCircle = OlGeomCircle_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeomLineString = OlGeomLineString_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeomMultiLineString = OlGeomMultiLineString_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeomMultiPoint = OlGeomMultiPoint_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeomMultiPolygon = OlGeomMultiPolygon_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeomPoint = OlGeomPoint_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGeomPolygon = OlGeomPolygon_vue_vue_type_script_setup_true_lang._sfc_main;
exports.default = install;
exports.install = install;
