// vue3-openlayers/dist/cjs/layers/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const layers_LayersCommonProps = require("./LayersCommonProps.js");
const OlAnimatedClusterLayer_vue_vue_type_script_setup_true_lang = require("../OlAnimatedClusterLayer.vue_vue_type_script_setup_true_lang-CCNEqQ9i.js");
const OlGraticuleLayer_vue_vue_type_script_setup_true_lang = require("../OlGraticuleLayer.vue_vue_type_script_setup_true_lang-Cng_AiRH.js");
const OlHeatmapLayer_vue_vue_type_script_setup_true_lang = require("../OlHeatmapLayer.vue_vue_type_script_setup_true_lang-B9zedv5v.js");
const OlImageLayer_vue_vue_type_script_setup_true_lang = require("../OlImageLayer.vue_vue_type_script_setup_true_lang-B8f2nWMu.js");
const OlLayerGroup_vue_vue_type_script_setup_true_lang = require("../OlLayerGroup.vue_vue_type_script_setup_true_lang-CaMan6aW.js");
const OlTileLayer_vue_vue_type_script_setup_true_lang = require("../OlTileLayer.vue_vue_type_script_setup_true_lang-BInKQGek.js");
const OlVectorLayer_vue_vue_type_script_setup_true_lang = require("../OlVectorLayer.vue_vue_type_script_setup_true_lang-BK1zICV1.js");
const OlVectorTileLayer_vue_vue_type_script_setup_true_lang = require("../OlVectorTileLayer.vue_vue_type_script_setup_true_lang-Cc_S7a6J.js");
const OlVectorImageLayer_vue_vue_type_script_setup_true_lang = require("../OlVectorImageLayer.vue_vue_type_script_setup_true_lang-Db7Rw_jx.js");
const OlWebglTileLayer_vue_vue_type_script_setup_true_lang = require("../OlWebglTileLayer.vue_vue_type_script_setup_true_lang-DCB8vbWI.js");
const OlWebglVectorLayer_vue_vue_type_script_setup_true_lang = require("../OlWebglVectorLayer.vue_vue_type_script_setup_true_lang-Cry9xLh6.js");
const registerWithAliases = require("../registerWithAliases-CfHnojZU.js");
function install(app, options) {
  registerWithAliases.registerWithAliases(app, "OlAnimatedClusterLayer", OlAnimatedClusterLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlGraticuleLayer", OlGraticuleLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlHeatmapLayer", OlHeatmapLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlImageLayer", OlImageLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlLayerGroup", OlLayerGroup_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlTileLayer", OlTileLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlVectorImageLayer", OlVectorImageLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlVectorLayer", OlVectorLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlVectorTileLayer", OlVectorTileLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlWebglTileLayer", OlWebglTileLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlWebglVectorLayer", OlWebglVectorLayer_vue_vue_type_script_setup_true_lang._sfc_main);
  if (options) {
    app.provide("ol-options", options);
  }
}
exports.layersCommonDefaultProps = layers_LayersCommonProps.layersCommonDefaultProps;
exports.useLayerDefaults = layers_LayersCommonProps.useDefaults;
exports.OlAnimatedClusterLayer = OlAnimatedClusterLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlGraticuleLayer = OlGraticuleLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlHeatmapLayer = OlHeatmapLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlImageLayer = OlImageLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlLayerGroup = OlLayerGroup_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlTileLayer = OlTileLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlVectorLayer = OlVectorLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlVectorTileLayer = OlVectorTileLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlVectorImageLayer = OlVectorImageLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlWebglTileLayer = OlWebglTileLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlWebglVectorLayer = OlWebglVectorLayer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.default = install;
exports.install = install;
