// vue3-openlayers/dist/esm/controls/index.js
import { _ as _sfc_main } from "../OlAttributionControl.vue_vue_type_script_setup_true_lang-Ch-Se63f.js";
import { _ as _sfc_main$e } from "../OlButtonControl.vue_vue_type_script_setup_true_lang-BGlYbGe0.js";
import { _ as _sfc_main$9 } from "../OlContextMenuControl.vue_vue_type_script_setup_true_lang-BPm6R-AQ.js";
import { _ as _sfc_main$c } from "../OlControlBar.vue_vue_type_script_setup_true_lang-Ceux-HyR.js";
import { _ as _sfc_main$1 } from "../OlFullScreenControl.vue_vue_type_script_setup_true_lang-C_U0swSk.js";
import { _ as _sfc_main$i } from "../OlLayerSwitcherControl.vue_vue_type_script_setup_true_lang-zg9ou0YU.js";
import { _ as _sfc_main$j } from "../OlLayerSwitcherImageControl.vue_vue_type_script_setup_true_lang-BZoTFgp8.js";
import { _ as _sfc_main$2 } from "../OlMousePositionControl.vue_vue_type_script_setup_true_lang-CK_BZWpg.js";
import { _ as _sfc_main$3 } from "../OlOverviewMapControl.vue_vue_type_script_setup_true_lang-DjSUh1ML.js";
import { _ as _sfc_main$f } from "../OlPrintDialogControl.vue_vue_type_script_setup_true_lang-wlEe8Ge2.js";
import { _ as _sfc_main$g } from "../OlProfileControl.vue_vue_type_script_setup_true_lang-Cz-5J71K.js";
import { _ as _sfc_main$8 } from "../OlRotateControl.vue_vue_type_script_setup_true_lang-DToHvbgv.js";
import { _ as _sfc_main$4 } from "../OlScaleLineControl.vue_vue_type_script_setup_true_lang-Ct3p9GVR.js";
import { _ as _sfc_main$a } from "../OlSearchControl.vue_vue_type_script_setup_true_lang-90dpSiOb.js";
import { _ as _sfc_main$b } from "../OlSwipeControl.vue_vue_type_script_setup_true_lang-CU-6IPsC.js";
import { _ as _sfc_main$d } from "../OlToggleControl.vue_vue_type_script_setup_true_lang-BctgQeYP.js";
import { _ as _sfc_main$h } from "../OlVideoRecorderControl.vue_vue_type_script_setup_true_lang-BoAdojYw.js";
import { _ as _sfc_main$k } from "../OlZoneControl.vue_vue_type_script_setup_true_lang-CIKm3j5y.js";
import { _ as _sfc_main$5 } from "../OlZoomControl.vue_vue_type_script_setup_true_lang-BnYSXUto.js";
import { _ as _sfc_main$6 } from "../OlZoomSliderControl.vue_vue_type_script_setup_true_lang-BNRRdnKP.js";
import { _ as _sfc_main$7 } from "../OlZoomToExtentControl.vue_vue_type_script_setup_true_lang-BdVoCvbJ.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlAttributionControl", _sfc_main);
  registerWithAliases(app, "OlFullScreenControl", _sfc_main$1);
  registerWithAliases(app, "OlMousePositionControl", _sfc_main$2);
  registerWithAliases(app, "OlOverviewMapControl", _sfc_main$3);
  registerWithAliases(app, "OlScaleLineControl", _sfc_main$4);
  registerWithAliases(app, "OlZoomControl", _sfc_main$5);
  registerWithAliases(app, "OlZoomSliderControl", _sfc_main$6);
  registerWithAliases(app, "OlZoomToExtentControl", _sfc_main$7);
  registerWithAliases(app, "OlRotateControl", _sfc_main$8);
  registerWithAliases(app, "OlContextMenuControl", _sfc_main$9);
  registerWithAliases(app, "OlSearchControl", _sfc_main$a);
  registerWithAliases(app, "OlSwipeControl", _sfc_main$b);
  registerWithAliases(app, "OlControlBar", _sfc_main$c);
  registerWithAliases(app, "OlToggleControl", _sfc_main$d);
  registerWithAliases(app, "OlButtonControl", _sfc_main$e);
  registerWithAliases(app, "OlPrintDialogControl", _sfc_main$f);
  registerWithAliases(app, "OlProfileControl", _sfc_main$g);
  registerWithAliases(app, "OlVideoRecorderControl", _sfc_main$h);
  registerWithAliases(app, "OlLayerSwitcherControl", _sfc_main$i);
  registerWithAliases(
    app,
    "OlLayerSwitcherImageControl",
    _sfc_main$j
  );
  registerWithAliases(app, "OlZoneControl", _sfc_main$k);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlAttributionControl,
  _sfc_main$e as OlButtonControl,
  _sfc_main$9 as OlContextMenuControl,
  _sfc_main$c as OlControlBar,
  _sfc_main$1 as OlFullScreenControl,
  _sfc_main$i as OlLayerSwitcherControl,
  _sfc_main$j as OlLayerSwitcherImageControl,
  _sfc_main$2 as OlMousePositionControl,
  _sfc_main$3 as OlOverviewMapControl,
  _sfc_main$f as OlPrintDialogControl,
  _sfc_main$g as OlProfileControl,
  _sfc_main$8 as OlRotateControl,
  _sfc_main$4 as OlScaleLineControl,
  _sfc_main$a as OlSearchControl,
  _sfc_main$b as OlSwipeControl,
  _sfc_main$d as OlToggleControl,
  _sfc_main$h as OlVideoRecorderControl,
  _sfc_main$k as OlZoneControl,
  _sfc_main$5 as OlZoomControl,
  _sfc_main$6 as OlZoomSliderControl,
  _sfc_main$7 as OlZoomToExtentControl,
  install as default,
  install
};
