// vue3-openlayers/dist/esm/geometries/index.js
import { _ as _sfc_main } from "../OlGeomCircle.vue_vue_type_script_setup_true_lang-qJ7x_BVN.js";
import { _ as _sfc_main$1 } from "../OlGeomLineString.vue_vue_type_script_setup_true_lang-IpW-OR_5.js";
import { _ as _sfc_main$2 } from "../OlGeomMultiLineString.vue_vue_type_script_setup_true_lang-B5X7eLSw.js";
import { _ as _sfc_main$3 } from "../OlGeomMultiPoint.vue_vue_type_script_setup_true_lang-CeZhF4p9.js";
import { _ as _sfc_main$4 } from "../OlGeomMultiPolygon.vue_vue_type_script_setup_true_lang-B8KYLhR3.js";
import { _ as _sfc_main$5 } from "../OlGeomPoint.vue_vue_type_script_setup_true_lang-bzZ1l_oY.js";
import { _ as _sfc_main$6 } from "../OlGeomPolygon.vue_vue_type_script_setup_true_lang-V5fcFurJ.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlGeomCircle", _sfc_main);
  registerWithAliases(app, "OlGeomLineString", _sfc_main$1);
  registerWithAliases(app, "OlGeomMultiLineString", _sfc_main$2);
  registerWithAliases(app, "OlGeomMultiPoint", _sfc_main$3);
  registerWithAliases(app, "OlGeomMultiPolygon", _sfc_main$4);
  registerWithAliases(app, "OlGeomPoint", _sfc_main$5);
  registerWithAliases(app, "OlGeomPolygon", _sfc_main$6);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlGeomCircle,
  _sfc_main$1 as OlGeomLineString,
  _sfc_main$2 as OlGeomMultiLineString,
  _sfc_main$3 as OlGeomMultiPoint,
  _sfc_main$4 as OlGeomMultiPolygon,
  _sfc_main$5 as OlGeomPoint,
  _sfc_main$6 as OlGeomPolygon,
  install as default,
  install
};
