// vue3-openlayers/dist/cjs/styles/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const OlStyle_vue_vue_type_script_setup_true_lang = require("../OlStyle.vue_vue_type_script_setup_true_lang-DIafFSnU.js");
const OlStyleCircle_vue_vue_type_script_setup_true_lang = require("../OlStyleCircle.vue_vue_type_script_setup_true_lang-BkMfTzVy.js");
const OlStyleStroke_vue_vue_type_script_setup_true_lang = require("../OlStyleStroke.vue_vue_type_script_setup_true_lang-T9R5s0Xw.js");
const OlStyleFill_vue_vue_type_script_setup_true_lang = require("../OlStyleFill.vue_vue_type_script_setup_true_lang-CXjlO0Sl.js");
const OlStyleIcon_vue_vue_type_script_setup_true_lang = require("../OlStyleIcon.vue_vue_type_script_setup_true_lang-BrhOSJGv.js");
const OlStyleText_vue_vue_type_script_setup_true_lang = require("../OlStyleText.vue_vue_type_script_setup_true_lang-DdQ9mKeQ.js");
const OlStyleFlowLine_vue_vue_type_script_setup_true_lang = require("../OlStyleFlowLine.vue_vue_type_script_setup_true_lang-I4sbfl4Q.js");
const registerWithAliases = require("../registerWithAliases-CfHnojZU.js");
function install(app, options) {
  registerWithAliases.registerWithAliases(app, "OlStyle", OlStyle_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlStyleCircle", OlStyleCircle_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlStyleStroke", OlStyleStroke_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlStyleFill", OlStyleFill_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlStyleIcon", OlStyleIcon_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlStyleText", OlStyleText_vue_vue_type_script_setup_true_lang._sfc_main);
  registerWithAliases.registerWithAliases(app, "OlStyleFlowLine", OlStyleFlowLine_vue_vue_type_script_setup_true_lang._sfc_main);
  if (options) {
    app.provide("ol-options", options);
  }
}
exports.OlStyle = OlStyle_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlStyleCircle = OlStyleCircle_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlStyleStroke = OlStyleStroke_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlStyleFill = OlStyleFill_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlStyleIcon = OlStyleIcon_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlStyleText = OlStyleText_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlStyleFlowLine = OlStyleFlowLine_vue_vue_type_script_setup_true_lang._sfc_main;
exports.default = install;
exports.install = install;
