// vue3-openlayers/dist/cjs/interactions/index.js
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const OlInteractionClusterSelect_vue_vue_type_script_setup_true_lang = require("../OlInteractionClusterSelect.vue_vue_type_script_setup_true_lang-8RxjzNyt.js");
const OlInteractionDragBox_vue_vue_type_script_setup_true_lang = require("../OlInteractionDragBox.vue_vue_type_script_setup_true_lang-C4cAzdL9.js");
const OlInteractionDragRotate_vue_vue_type_script_setup_true_lang = require("../OlInteractionDragRotate.vue_vue_type_script_setup_true_lang-BS1uXXCJ.js");
const OlInteractionDragRotateAndZoom_vue_vue_type_script_setup_true_lang = require("../OlInteractionDragRotateAndZoom.vue_vue_type_script_setup_true_lang-BVgvRWBy.js");
const OlInteractionLink_vue_vue_type_script_setup_true_lang = require("../OlInteractionLink.vue_vue_type_script_setup_true_lang-Ca509Yoe.js");
const OlInteractionSelect_vue_vue_type_script_setup_true_lang = require("../OlInteractionSelect.vue_vue_type_script_setup_true_lang-Cqf2HRmg.js");
const OlInteractionDraw_vue_vue_type_script_setup_true_lang = require("../OlInteractionDraw.vue_vue_type_script_setup_true_lang-oyDYpcig.js");
const OlInteractionModify_vue_vue_type_script_setup_true_lang = require("../OlInteractionModify.vue_vue_type_script_setup_true_lang-p9VlQj3f.js");
const OlInteractionPointer_vue_vue_type_script_setup_true_lang = require("../OlInteractionPointer.vue_vue_type_script_setup_true_lang-ARMdDoXt.js");
const OlInteractionSnap_vue_vue_type_script_setup_true_lang = require("../OlInteractionSnap.vue_vue_type_script_setup_true_lang-BbU-hqZk.js");
const OlInteractionTransform_vue_vue_type_script_setup_true_lang = require("../OlInteractionTransform.vue_vue_type_script_setup_true_lang-Dl-9_ud1.js");
const OlInteractionMouseWheelZoom_vue_vue_type_script_setup_true_lang = require("../OlInteractionMouseWheelZoom.vue_vue_type_script_setup_true_lang-Ce_URBCE.js");
const OlInteractionSynchronize_vue_vue_type_script_setup_true_lang = require("../OlInteractionSynchronize.vue_vue_type_script_setup_true_lang-D4P4Qllv.js");
function install(app, options) {
  app.component("OlInteractionClusterSelect", OlInteractionClusterSelect_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionDragBox", OlInteractionDragBox_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionDragRotate", OlInteractionDragRotate_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component(
    "OlInteractionDragRotateAndZoom",
    OlInteractionDragRotateAndZoom_vue_vue_type_script_setup_true_lang._sfc_main
  );
  app.component("OlInteractionLink", OlInteractionLink_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionSelect", OlInteractionSelect_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionDraw", OlInteractionDraw_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionModify", OlInteractionModify_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionPointer", OlInteractionPointer_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionSnap", OlInteractionSnap_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionTransform", OlInteractionTransform_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionMouseWheelZoom", OlInteractionMouseWheelZoom_vue_vue_type_script_setup_true_lang._sfc_main);
  app.component("OlInteractionSynchronize", OlInteractionSynchronize_vue_vue_type_script_setup_true_lang._sfc_main);
  if (options) {
    app.provide("ol-options", options);
  }
}
exports.OlInteractionClusterSelect = OlInteractionClusterSelect_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionDragBox = OlInteractionDragBox_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionDragRotate = OlInteractionDragRotate_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionDragRotateAndZoom = OlInteractionDragRotateAndZoom_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionLink = OlInteractionLink_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionSelect = OlInteractionSelect_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionDraw = OlInteractionDraw_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionModify = OlInteractionModify_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionPointer = OlInteractionPointer_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionSnap = OlInteractionSnap_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionTransform = OlInteractionTransform_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionMouseWheelZoom = OlInteractionMouseWheelZoom_vue_vue_type_script_setup_true_lang._sfc_main;
exports.OlInteractionSynchronize = OlInteractionSynchronize_vue_vue_type_script_setup_true_lang._sfc_main;
exports.default = install;
exports.install = install;
