// vue3-openlayers/dist/cjs/OlAnimatedClusterLayer.vue_vue_type_script_setup_true_lang-CCNEqQ9i.js
"use strict";
const vue = require("vue");
const source = require("ol/source");
const AnimatedCluster = require("ol-ext/layer/AnimatedCluster");
const composables_useLayer = require("./composables/useLayer.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlAnimatedClusterLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    distance: {},
    animationDuration: {},
    easingFunction: { type: Function },
    animationMethod: { type: Function },
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    renderOrder: { type: Function },
    renderBuffer: {},
    source: {},
    map: {},
    declutter: { type: [String, Number, Boolean] },
    style: { type: [Object, Array, Function, null] },
    background: { type: [String, Function] },
    updateWhileAnimating: { type: Boolean },
    updateWhileInteracting: { type: Boolean },
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, layers_LayersCommonProps.useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "addfeature", "changefeature", "clear", "featuresloadend", "featuresloaderror", "featuresloadstart", "removefeature"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const clusterSource = vue.shallowRef(
      new source.Cluster({
        distance: props.distance,
        geometryFunction: (feature) => feature.getGeometry()
      })
    );
    const { layer, map } = composables_useLayer.default(
      AnimatedCluster,
      {
        ...props,
        source: clusterSource.value
      },
      composables_useOpenLayersEvents.VECTOR_SOURCE_EVENTS
    );
    vue.watch(
      () => props.distance,
      (newValue) => {
        clusterSource.value.setDistance(newValue);
      }
    );
    vue.provide("vectorLayer", clusterSource);
    vue.provide("stylable", layer);
    __expose({
      vectorLayer: layer,
      map
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
