// vue3-openlayers/dist/cjs/OlAttributionControl.vue_vue_type_script_setup_true_lang-CYOdZrBY.js
"use strict";
const vue = require("vue");
const Attribution = require("ol/control/Attribution");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlAttributionControl",
  props: {
    className: {},
    target: {},
    collapsible: { type: Boolean },
    collapsed: { type: Boolean, default: true },
    tipLabel: {},
    label: {},
    expandClassName: {},
    collapseLabel: {},
    collapseClassName: {},
    render: {},
    attributions: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Attribution, properties, attrs);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
