// vue3-openlayers/dist/cjs/OlButtonControl.vue_vue_type_script_setup_true_lang-BUVSEqU3.js
"use strict";
const vue = require("vue");
const Button = require("ol-ext/control/Button");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlButtonControl",
  props: {
    className: {},
    classButton: {},
    title: {},
    name: {},
    html: {},
    handleClick: { type: Function },
    attributes: {},
    element: {},
    render: { type: Function },
    target: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Button, properties, attrs);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
