// vue3-openlayers/dist/cjs/OlContextMenuControl.vue_vue_type_script_setup_true_lang-kXrrZ6Ro.js
"use strict";
const vue = require("vue");
const ContextMenu = require("ol-contextmenu");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlContextMenuControl",
  props: {
    width: { default: 150 },
    scrollAt: {},
    eventType: { default: "contextmenu" },
    defaultItems: { type: Boolean, default: true },
    items: { default: () => [] }
  },
  emits: ["change", "error", "propertychange", "beforeopen", "open", "close"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(ContextMenu, properties, attrs, [
      "beforeopen",
      "open",
      "close"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
