// vue3-openlayers/dist/cjs/OlControlBar.vue_vue_type_script_setup_true_lang-DjIAEAXc.js
"use strict";
const vue = require("vue");
const Bar = require("ol-ext/control/Bar");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlControlBar",
  props: {
    className: {},
    group: { type: Boolean, default: true },
    toggleOne: { type: Boolean, default: true },
    autoDeactivate: { type: Boolean },
    controls: {},
    attributes: {},
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Bar, properties, attrs);
    vue.provide("controlBar", control);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
