// vue3-openlayers/dist/cjs/OlFullScreenControl.vue_vue_type_script_setup_true_lang-DnP_M9aQ.js
"use strict";
const vue = require("vue");
const FullScreen = require("ol/control/FullScreen");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlFullScreenControl",
  props: {
    className: {},
    label: {},
    labelActive: {},
    activeClassName: {},
    inactiveClassName: {},
    tipLabel: {},
    keys: { type: Boolean },
    target: {},
    source: {}
  },
  emits: ["change", "error", "propertychange", "enterfullscreen", "leavefullscreen"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(FullScreen, properties, attrs, [
      "enterfullscreen",
      "leavefullscreen"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
