// vue3-openlayers/dist/cjs/OlGeoLocation.vue_vue_type_script_setup_true_lang-Bj7vrkKX.js
"use strict";
const vue = require("vue");
const Geolocation = require("ol/Geolocation");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlGeoLocation",
  props: {
    tracking: { type: Boolean, default: true },
    trackingOptions: { default: () => ({
      enableHighAccuracy: true
    }) },
    projection: { default: "EPSG:3857" }
  },
  emits: ["change", "error", "propertychange", "change:accuracy", "change:accuracyGeometry", "change:altitude", "change:altitudeAccuracy", "change:heading", "change:position", "change:projection", "change:speed", "change:tracking", "change:trackingOptions"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const properties = composables_usePropsAsObjectProperties.default(props);
    const geoLoc = vue.computed(() => new Geolocation(properties));
    composables_useOpenLayersEvents.useOpenLayersEvents(geoLoc, [
      "change:accuracy",
      "change:accuracyGeometry",
      "change:altitude",
      "change:altitudeAccuracy",
      "change:heading",
      "change:position",
      "change:projection",
      "change:speed",
      "change:tracking",
      "change:trackingOptions"
    ]);
    const position = vue.ref([]);
    const accuracy = vue.ref(0);
    const altitude = vue.ref(0);
    const altitudeAccuracy = vue.ref(0);
    const speed = vue.ref(0);
    const heading = vue.ref(0);
    const accuracyGeometry = vue.ref(null);
    __expose({
      geoLoc,
      position,
      accuracy,
      altitude,
      altitudeAccuracy,
      speed,
      heading,
      accuracyGeometry
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default", {
        position: position.value,
        speed: speed.value,
        heading: heading.value,
        accuracy: accuracy.value,
        altitude: altitude.value,
        altitudeAccuracy: altitudeAccuracy.value,
        accuracyGeometry: accuracyGeometry.value
      });
    };
  }
});
exports._sfc_main = _sfc_main;
