// vue3-openlayers/dist/cjs/OlGraticuleLayer.vue_vue_type_script_setup_true_lang-Cng_AiRH.js
"use strict";
const vue = require("vue");
const Graticule = require("ol/layer/Graticule");
const composables_useLayer = require("./composables/useLayer.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlGraticuleLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    maxLines: {},
    strokeStyle: {},
    targetSize: {},
    showLabels: { type: Boolean },
    lonLabelFormatter: { type: Function },
    latLabelFormatter: { type: Function },
    lonLabelPosition: {},
    latLabelPosition: {},
    lonLabelStyle: {},
    latLabelStyle: {},
    intervals: {},
    wrapX: { type: Boolean },
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, layers_LayersCommonProps.useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = composables_useLayer.default(Graticule, props);
    vue.provide("vectorLayer", layer);
    vue.provide("stylable", layer);
    __expose({
      vectorLayer: layer
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
