// vue3-openlayers/dist/cjs/OlHeatmapLayer.vue_vue_type_script_setup_true_lang-B9zedv5v.js
"use strict";
const vue = require("vue");
const HeatmapLayer = require("ol/layer/Heatmap");
const composables_useLayer = require("./composables/useLayer.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlHeatmapLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    source: {},
    map: {},
    render: { type: Function },
    properties: {},
    gradient: {},
    radius: {},
    blur: {},
    weight: { type: [Number, Array, String, Function] },
    filter: { type: [Boolean, Array] },
    variables: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, layers_LayersCommonProps.useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "change:blur", "change:gradient", "change:radius"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = composables_useLayer.default(HeatmapLayer, props, [
      "change:blur",
      "change:extent",
      "change:gradient",
      "change:radius"
    ]);
    vue.provide("heatmapLayer", layer);
    vue.provide("stylable", layer);
    __expose({
      heatmapLayer: layer
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
