// vue3-openlayers/dist/cjs/OlImageLayer.vue_vue_type_script_setup_true_lang-B8f2nWMu.js
"use strict";
const vue = require("vue");
const ImageLayer = require("ol/layer/Image");
const composables_useLayer = require("./composables/useLayer.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlImageLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    map: {},
    source: {},
    properties: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, layers_LayersCommonProps.useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = composables_useLayer.default(ImageLayer, props);
    vue.provide("imageLayer", layer);
    __expose({
      imageLayer: layer
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
