// vue3-openlayers/dist/cjs/OlInteractionDragBox.vue_vue_type_script_setup_true_lang-C4cAzdL9.js
"use strict";
const vue = require("vue");
const DragBox = require("ol/interaction/DragBox");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionDragBox",
  props: {
    className: {},
    condition: { type: Function },
    minArea: {},
    boxEndCondition: { type: Function },
    onBoxEnd: { type: Function }
  },
  emits: ["change", "error", "propertychange", "boxcancel", "boxdrag", "boxend", "boxstart"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const dragbox = vue.computed(() => new DragBox(properties));
    composables_useOpenLayersEvents.useOpenLayersEvents(dragbox, ["boxcancel", "boxdrag", "boxend", "boxstart"]);
    vue.watch(dragbox, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(dragbox.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(dragbox.value);
    });
    __expose({
      dragbox
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
