// vue3-openlayers/dist/cjs/OlInteractionSnap.vue_vue_type_script_setup_true_lang-BbU-hqZk.js
"use strict";
const vue = require("vue");
const Snap = require("ol/interaction/Snap");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlInteractionSnap",
  props: {
    features: {},
    source: {},
    edge: { type: Boolean, default: true },
    vertex: { type: Boolean, default: true },
    intersection: { type: Boolean },
    pixelTolerance: {},
    segmenters: {}
  },
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const vectorSource = vue.inject("vectorSource");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const createSnap = () => new Snap({
      ...properties,
      source: props.source || vectorSource?.value
    });
    let snap = createSnap();
    vue.watch(properties, () => {
      map?.removeInteraction(snap);
      snap = createSnap();
      map?.addInteraction(snap);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(snap);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(snap);
    });
    __expose({
      snap
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
