// vue3-openlayers/dist/cjs/OlInteractionClusterSelect.vue_vue_type_script_setup_true_lang-8RxjzNyt.js
"use strict";
const vue = require("vue");
const SelectCluster = require("ol-ext/interaction/SelectCluster");
const Style = require("ol/style/Style");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionClusterSelect",
  props: {
    featureStyle: {},
    selectCluster: { type: Boolean, default: true },
    pointRadius: { default: 7 },
    spiral: { type: Boolean, default: true },
    circleMaxObject: {},
    maxObjects: {},
    animate: { type: Boolean, default: true },
    animationDuration: { default: 500 },
    autoClose: { type: Boolean, default: true },
    addCondition: {},
    condition: {},
    layers: {},
    style: {},
    removeCondition: {},
    toggleCondition: {},
    multi: { type: Boolean, default: false },
    features: {},
    filter: {},
    hitTolerance: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const select = vue.computed(
      () => new SelectCluster({
        style: new Style(),
        ...properties
      })
    );
    composables_useOpenLayersEvents.useOpenLayersEvents(select, ["select"]);
    vue.watch(select, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(select.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(select.value);
    });
    vue.provide("stylable", select);
    __expose({
      select
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
