// vue3-openlayers/dist/cjs/OlInteractionMouseWheelZoom.vue_vue_type_script_setup_true_lang-Ce_URBCE.js
"use strict";
const vue = require("vue");
const MouseWheelZoom = require("ol/interaction/MouseWheelZoom");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlInteractionMouseWheelZoom",
  props: {
    condition: {},
    onFocusOnly: { type: Boolean },
    maxDelta: {},
    duration: {},
    timeout: {},
    useAnchor: { type: Boolean, default: true },
    constrainResolution: { type: Boolean }
  },
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const mouseWheelZoom = vue.computed(() => new MouseWheelZoom(properties));
    vue.watch(mouseWheelZoom, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.getInteractions().forEach((interaction) => {
        if (interaction instanceof MouseWheelZoom) {
          map?.removeInteraction(interaction);
        }
      });
      map?.addInteraction(mouseWheelZoom.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(mouseWheelZoom.value);
    });
    __expose({
      mouseWheelZoom
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
