// vue3-openlayers/dist/cjs/OlInteractionModify.vue_vue_type_script_setup_true_lang-p9VlQj3f.js
"use strict";
const vue = require("vue");
const Modify = require("ol/interaction/Modify");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionModify",
  props: {
    condition: { type: Function },
    deleteCondition: { type: Function },
    insertVertexCondition: { type: Function },
    pixelTolerance: {},
    style: { type: [Object, Array, Function] },
    source: {},
    hitDetection: { type: Boolean },
    features: {},
    wrapX: { type: Boolean },
    snapToPointer: { type: Boolean }
  },
  emits: ["change", "error", "propertychange", "modifystart", "modifyend"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const vectorSource = vue.inject("vectorSource", null);
    const {
      source,
      features,
      condition,
      deleteCondition,
      insertVertexCondition,
      pixelTolerance,
      wrapX,
      hitDetection,
      snapToPointer
    } = vue.toRefs(props);
    function createModify() {
      if (!(source?.value || features.value)) {
        console.error(
          `[Vue3-OpenLayers Error] OlInteractionModify: Modify interactions needs either a source or features to work.
      Please provide either the props 'source' or 'feature' or use the component with an '<OlSourceVector>' component.`
        );
      }
      return new Modify({
        source: source.value || vectorSource?.value,
        features: features.value,
        condition: condition.value,
        deleteCondition: deleteCondition.value,
        insertVertexCondition: insertVertexCondition.value,
        pixelTolerance: pixelTolerance.value,
        wrapX: wrapX.value,
        hitDetection: hitDetection.value,
        snapToPointer: snapToPointer.value
      });
    }
    let modify = createModify();
    composables_useOpenLayersEvents.useOpenLayersEvents(modify, ["modifystart", "modifyend"]);
    vue.watch(
      [
        condition,
        deleteCondition,
        insertVertexCondition,
        pixelTolerance,
        wrapX,
        hitDetection,
        snapToPointer
      ],
      () => {
        modify.dispose();
        map?.removeInteraction(modify);
        modify = createModify();
        map?.addInteraction(modify);
        map?.changed();
      }
    );
    vue.onMounted(() => {
      map?.addInteraction(modify);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(modify);
    });
    vue.provide("stylable", vue.ref(modify));
    __expose({
      modify
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
