// vue3-openlayers/dist/cjs/OlInteractionDraw.vue_vue_type_script_setup_true_lang-oyDYpcig.js
"use strict";
const vue = require("vue");
const Draw = require("ol/interaction/Draw");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionDraw",
  props: {
    type: {},
    clickTolerance: {},
    features: {},
    source: {},
    dragVertexDelay: {},
    snapTolerance: {},
    stopClick: { type: Boolean },
    maxPoints: {},
    minPoints: {},
    finishCondition: { type: Function },
    style: { type: [Object, Array, Function] },
    geometryFunction: { type: Function },
    geometryName: {},
    condition: { type: Function },
    freehand: { type: Boolean },
    freehandCondition: { type: Function },
    trace: { type: [Boolean, Function] },
    traceSource: {},
    wrapX: { type: Boolean },
    geometryLayout: {}
  },
  emits: ["change", "error", "propertychange", "drawstart", "drawend"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const vectorSource = vue.inject("vectorSource", null);
    const {
      source,
      type,
      clickTolerance,
      dragVertexDelay,
      snapTolerance,
      stopClick,
      maxPoints,
      minPoints,
      finishCondition,
      geometryFunction,
      geometryName,
      condition,
      freehand,
      freehandCondition,
      wrapX,
      features,
      geometryLayout,
      trace,
      traceSource
    } = vue.toRefs(props);
    function createDraw() {
      return new Draw({
        source: source.value || vectorSource?.value,
        type: type.value,
        clickTolerance: clickTolerance.value,
        dragVertexDelay: dragVertexDelay.value,
        snapTolerance: snapTolerance.value,
        stopClick: stopClick.value,
        maxPoints: maxPoints.value,
        minPoints: minPoints.value,
        finishCondition: finishCondition.value,
        geometryFunction: geometryFunction.value,
        geometryName: geometryName.value,
        condition: condition.value,
        freehand: freehand.value,
        freehandCondition: freehandCondition.value,
        wrapX: wrapX.value,
        features: features.value,
        geometryLayout: geometryLayout.value,
        trace: trace.value,
        traceSource: traceSource.value
      });
    }
    const draw = vue.shallowRef(createDraw());
    const { updateOpenLayersEventHandlers } = composables_useOpenLayersEvents.useOpenLayersEvents(draw, [
      "drawstart",
      "drawend"
    ]);
    vue.watch(
      [
        source,
        type,
        clickTolerance,
        dragVertexDelay,
        snapTolerance,
        stopClick,
        maxPoints,
        minPoints,
        finishCondition,
        geometryFunction,
        geometryName,
        condition,
        freehand,
        freehandCondition,
        wrapX,
        features,
        geometryLayout,
        trace,
        traceSource
      ],
      () => {
        draw.value.abortDrawing();
        map?.removeInteraction(draw.value);
        draw.value = createDraw();
        updateOpenLayersEventHandlers();
        map?.addInteraction(draw.value);
        map?.changed();
      }
    );
    vue.onMounted(() => {
      map?.addInteraction(draw.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(draw.value);
    });
    vue.provide("stylable", draw);
    __expose({
      draw: draw.value
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
