// vue3-openlayers/dist/cjs/OlInteractionLink.vue_vue_type_script_setup_true_lang-Ca509Yoe.js
"use strict";
const vue = require("vue");
const Link = require("ol/interaction/Link");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionLink",
  props: {
    animate: { type: [Boolean, Object], default: true },
    params: { default: () => ["x", "y", "z", "r", "l"] },
    replace: { type: Boolean, default: false },
    prefix: { default: "" }
  },
  emits: ["change", "error", "propertychange", "change:active"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const link = vue.shallowRef(new Link(properties));
    composables_useOpenLayersEvents.useOpenLayersEvents(link, ["change:active"]);
    vue.watch(link, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(link.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(link.value);
    });
    vue.provide("stylable", link);
    __expose({
      link
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
