// vue3-openlayers/dist/cjs/OlInteractionSelect.vue_vue_type_script_setup_true_lang-Cqf2HRmg.js
"use strict";
const vue = require("vue");
const Select = require("ol/interaction/Select");
const Style = require("ol/style/Style");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionSelect",
  props: {
    addCondition: { type: Function },
    condition: { type: Function },
    layers: { type: [Array, Function] },
    style: { type: [Object, Array, Function, null] },
    removeCondition: { type: Function },
    toggleCondition: { type: Function },
    multi: { type: Boolean },
    features: {},
    filter: { type: Function },
    hitTolerance: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const select = vue.computed(
      () => new Select({
        style: new Style(),
        ...properties
      })
    );
    composables_useOpenLayersEvents.useOpenLayersEvents(select, ["select"]);
    vue.watch(select, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(select.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(select.value);
    });
    vue.provide("stylable", select);
    __expose({
      select
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
