// vue3-openlayers/dist/cjs/OlInteractionTransform.vue_vue_type_script_setup_true_lang-Dl-9_ud1.js
"use strict";
const vue = require("vue");
const Transform = require("ol-ext/interaction/Transform");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const condition = require("ol/events/condition");
const functions = require("ol/functions");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionTransform",
  props: {
    filter: {},
    layers: {},
    features: {},
    addCondition: {},
    hitTolerance: {},
    translateFeature: { type: Boolean, default: true },
    translate: { type: Boolean, default: true },
    translateBBox: { type: Boolean },
    stretch: { type: Boolean, default: true },
    scale: { type: Boolean, default: true },
    rotate: { type: Boolean, default: true },
    noFlip: { type: Boolean },
    selection: { type: Boolean, default: functions.TRUE },
    keepAspectRatio: { type: Function, default: condition.always },
    modifyCenter: {},
    enableRotatedTransform: { type: Boolean },
    keepRectangle: { type: Boolean },
    style: {},
    pointRadius: {}
  },
  emits: ["change", "error", "propertychange", "select", "rotatestart", "rotating", "rotateend", "translatestart", "translating", "translateend", "scalestart", "scaling", "scaleend"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    function createTransform() {
      return new Transform(properties);
    }
    const transform = vue.shallowRef(createTransform());
    const { updateOpenLayersEventHandlers } = composables_useOpenLayersEvents.useOpenLayersEvents(transform, [
      "select",
      "rotatestart",
      "rotating",
      "rotateend",
      "translatestart",
      "translating",
      "translateend",
      "scalestart",
      "scaling",
      "scaleend"
    ]);
    vue.watch(transform, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      transform.value = createTransform();
      updateOpenLayersEventHandlers();
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(transform.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(transform.value);
    });
    vue.provide("stylable", transform);
    __expose({
      transform: transform.value
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
