// vue3-openlayers/dist/cjs/OlInteractionPointer.vue_vue_type_script_setup_true_lang-ARMdDoXt.js
"use strict";
const vue = require("vue");
const Pointer = require("ol/interaction/Pointer");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlInteractionPointer",
  props: {
    handleEvent: { type: Function },
    stopDown: { type: Function }
  },
  emits: ["change", "error", "propertychange", "down", "move", "up", "drag", "change:active"],
  setup(__props, { expose: __expose, emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const pointer = vue.shallowRef(
      new Pointer({
        ...properties,
        handleDownEvent: (e) => {
          emit("down", e);
          return true;
        },
        handleDragEvent: (e) => {
          emit("drag", e);
          return true;
        },
        handleMoveEvent: (e) => {
          emit("move", e);
        },
        handleUpEvent: (e) => {
          emit("up", e);
          return true;
        }
      })
    );
    composables_useOpenLayersEvents.useOpenLayersEvents(pointer, ["change:active"]);
    vue.watch(pointer, (newVal, oldVal) => {
      map?.removeInteraction(oldVal);
      map?.addInteraction(newVal);
      map?.changed();
    });
    vue.onMounted(() => {
      map?.addInteraction(pointer.value);
    });
    vue.onUnmounted(() => {
      map?.removeInteraction(pointer.value);
    });
    __expose({
      pointer
    });
    return (_ctx, _cache) => {
      return vue.renderSlot(_ctx.$slots, "default");
    };
  }
});
exports._sfc_main = _sfc_main;
