// vue3-openlayers/dist/cjs/OlLayerGroup.vue_vue_type_script_setup_true_lang-CaMan6aW.js
"use strict";
const vue = require("vue");
const LayerGroup = require("ol/layer/Group");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlLayerGroup",
  props: {
    opacity: { default: 1 },
    visible: { type: Boolean, default: true },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    layers: {},
    properties: { default: () => ({}) },
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean, default: true },
    preview: {}
  },
  emits: ["change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "change:layers"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const properties = composables_usePropsAsObjectProperties.default(props);
    const layerGroup = vue.shallowRef(new LayerGroup(properties));
    composables_useOpenLayersEvents.useOpenLayersEvents(layerGroup, [
      "change:extent",
      "change:layers",
      "change:maxResolution",
      "change:maxZoom",
      "change:minResolution",
      "change:minZoom",
      "change:opacity",
      "change:visible",
      "change:zIndex"
    ]);
    const parentLayerGroup = vue.inject("layerGroup", null);
    vue.watch(
      () => properties,
      (newValue) => {
        for (const key in newValue) {
          const keyInObj = key;
          if (newValue[keyInObj] !== void 0) {
            layerGroup.value.set(key, newValue[keyInObj]);
          }
        }
      },
      { deep: true }
    );
    vue.onMounted(() => {
      map?.addLayer(layerGroup.value);
      if (parentLayerGroup) {
        const layerCollection = parentLayerGroup.getLayers();
        layerCollection.push(layerGroup.value);
        parentLayerGroup.setLayers(layerCollection);
      }
    });
    vue.onUnmounted(() => {
      map?.removeLayer(layerGroup.value);
    });
    vue.provide("layerGroup", layerGroup.value);
    __expose({
      layerGroup: layerGroup.value
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
