// vue3-openlayers/dist/cjs/OlLayerSwitcherImageControl.vue_vue_type_script_setup_true_lang-DtCifgV-.js
"use strict";
const vue = require("vue");
const LayerSwitcherImage = require("ol-ext/control/LayerSwitcherImage");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlLayerSwitcherImageControl",
  props: {
    displayInLayerSwitcher: {},
    show_progress: { type: Boolean },
    mouseover: { type: Boolean },
    reordering: { type: Boolean, default: true },
    trash: { type: Boolean },
    oninfo: {},
    extent: { type: Boolean },
    onExtent: {},
    drawDelay: {},
    collapsed: { type: Boolean, default: true },
    layerGroup: {},
    noScroll: { type: Boolean },
    onchangeCheck: {},
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange", "drawlist", "layer:visible", "layer:opacity", "toggle", "reorder-start", "reorder-end"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(LayerSwitcherImage, properties, attrs, [
      "drawlist",
      "layer:visible",
      "layer:opacity",
      "toggle",
      "reorder-start",
      "reorder-end"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
