// vue3-openlayers/dist/cjs/OlMap.vue_vue_type_script_setup_true_lang-CO-brzHY.js
"use strict";
const vue = require("vue");
const Map = require("ol/Map");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const helpers_properties = require("./helpers/properties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlMap",
  props: {
    controls: {},
    pixelRatio: {},
    interactions: {},
    keyboardEventTarget: {},
    layers: {},
    maxTilesLoading: {},
    moveTolerance: {},
    overlays: {},
    target: {},
    view: {},
    instance: {}
  },
  emits: ["change", "error", "propertychange", "change:layerGroup", "change:size", "change:target", "change:view", "click", "dblclick", "singleclick", "loadstart", "loadend", "pointerdrag", "pointermove", "movestart", "moveend", "postrender", "precompose", "postcompose", "rendercomplete"],
  setup(__props, { expose: __expose, emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const properties = composables_usePropsAsObjectProperties.default({
      ...props,
      instance: void 0
    });
    const mapRef = vue.ref(void 0);
    let map = props.instance || new Map({ ...properties });
    vue.watch(
      properties,
      () => {
        const p = props.instance ? helpers_properties.mergeProperties(properties, props.instance.getProperties()) : properties;
        map?.setProperties(p);
      },
      { immediate: true }
    );
    vue.onMounted(() => {
      if (!props.instance) {
        map?.setTarget(mapRef.value);
      }
    });
    vue.onBeforeUnmount(() => {
      if (!props.instance) {
        map?.setTarget(void 0);
      }
    });
    vue.onUnmounted(() => {
      map = void 0;
    });
    vue.provide("map", map);
    const forEachFeatureAtPixel = (pixel, callback, options) => map?.forEachFeatureAtPixel(pixel, callback, options);
    const getCoordinateFromPixel = (pixel) => map?.getCoordinateFromPixel(pixel);
    const render = () => map?.render();
    const updateSize = () => map?.updateSize();
    map.on("change", (event) => emit("change", event));
    map.on("error", (event) => emit("error", event));
    map.on("propertychange", (event) => emit("propertychange", event));
    map.on("click", (event) => emit("click", event));
    map.on("change:size", (event) => emit("change:size", event));
    map.on("change:target", (event) => emit("change:target", event));
    map.on("change:view", (event) => emit("change:view", event));
    map.on("dblclick", (event) => emit("dblclick", event));
    map.on("singleclick", (event) => emit("singleclick", event));
    map.on("pointerdrag", (event) => emit("pointerdrag", event));
    map.on("pointermove", (event) => emit("pointermove", event));
    map.on("movestart", (event) => emit("movestart", event));
    map.on("moveend", (event) => emit("moveend", event));
    map.on("postrender", (event) => emit("postrender", event));
    map.on("precompose", (event) => emit("precompose", event));
    map.on("postcompose", (event) => emit("postcompose", event));
    map.on("rendercomplete", (event) => emit("rendercomplete", event));
    map.on("loadstart", (event) => {
      map?.getTargetElement().classList.add("ol-map");
      map?.getTargetElement().classList.add("ol-map-loading");
      map?.getTargetElement().classList.remove("ol-map-fully-loaded");
      emit("loadstart", event);
    });
    map.on("loadend", (event) => {
      map?.getTargetElement().classList.add("ol-map-fully-loaded");
      map?.getTargetElement().classList.remove("ol-map-loading");
      emit("loadend", event);
    });
    __expose({
      map,
      mapRef,
      forEachFeatureAtPixel,
      getCoordinateFromPixel,
      render,
      updateSize
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        ref_key: "mapRef",
        ref: mapRef
      }, [
        vue.renderSlot(_ctx.$slots, "default")
      ], 512);
    };
  }
});
exports._sfc_main = _sfc_main;
