// vue3-openlayers/dist/cjs/OlMousePositionControl.vue_vue_type_script_setup_true_lang-Bo2IiKI4.js
"use strict";
const vue = require("vue");
const MousePosition = require("ol/control/MousePosition");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlMousePositionControl",
  props: {
    className: {},
    coordinateFormat: {},
    projection: {},
    render: {},
    target: {},
    placeholder: {},
    wrapX: { type: Boolean, default: true }
  },
  emits: ["change", "error", "propertychange", "change:coordinateFormat", "change:projection"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(MousePosition, properties, attrs, [
      "change:coordinateFormat",
      "change:projection"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
