// vue3-openlayers/dist/cjs/OlOverviewMapControl.vue_vue_type_script_setup_true_lang-CZrM5Bht.js
"use strict";
const vue = require("vue");
const OverviewMap = require("ol/control/OverviewMap");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlOverviewMapControl",
  props: {
    className: {},
    collapsed: { type: Boolean, default: true },
    collapseLabel: {},
    collapsible: { type: Boolean, default: true },
    label: {},
    layers: {},
    render: {},
    rotateWithView: { type: Boolean },
    target: {},
    tipLabel: {},
    view: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(OverviewMap, properties, attrs);
    vue.provide("overviewMap", control);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
