// vue3-openlayers/dist/cjs/OlOverlay.vue_vue_type_script_setup_true_lang-C1JIynKY.js
"use strict";
const vue = require("vue");
const Overlay = require("ol/Overlay");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlOverlay",
  props: {
    id: {},
    element: {},
    offset: {},
    position: {},
    positioning: {},
    stopEvent: { type: Boolean, default: true },
    insertFirst: { type: Boolean, default: true },
    autoPan: { type: [Boolean, Object] },
    className: {}
  },
  emits: ["change", "error", "propertychange", "change:element", "change:map", "change:offset", "change:position", "change:positioning"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const map = vue.inject("map");
    const htmlContent = vue.ref();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const overlay = vue.shallowRef(new Overlay(properties));
    composables_useOpenLayersEvents.useOpenLayersEvents(overlay, [
      "change:element",
      "change:map",
      "change:offset",
      "change:position",
      "change:positioning"
    ]);
    const getOffset = () => overlay.value.getOffset();
    const getPosition = () => overlay.value.getPosition();
    const getPositioning = () => overlay.value.getPositioning();
    const panIntoView = (optPanIntoViewOptions) => overlay.value.panIntoView(optPanIntoViewOptions);
    const setElement = (element) => overlay.value.setElement(element);
    const setOffset = (offset) => overlay.value.setOffset(offset);
    const setPosition = (position) => overlay.value.setPosition(position);
    const setPositioning = (positioning) => overlay.value.setPositioning(positioning);
    function removeOverlay(ov) {
      const removed = map?.removeOverlay(ov);
      if (!removed) {
        console.warn("couldn't find matching overlay to remove", overlay.value);
      }
    }
    vue.onMounted(() => {
      map?.addOverlay(overlay.value);
    });
    vue.onUnmounted(() => removeOverlay(overlay.value));
    vue.watch(
      () => properties,
      (newValue) => {
        for (const key in newValue) {
          const keyInObj = key;
          if (newValue[keyInObj] !== void 0) {
            overlay.value.set(key, newValue[keyInObj]);
          }
        }
      },
      { deep: true }
    );
    vue.watchEffect(
      () => {
        setElement(htmlContent.value);
      },
      {
        flush: "post"
      }
    );
    __expose({
      overlay,
      htmlContent,
      getOffset,
      getPosition,
      getPositioning,
      panIntoView,
      setElement,
      setOffset,
      setPosition,
      setPositioning
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        ref_key: "htmlContent",
        ref: htmlContent
      }, [
        vue.renderSlot(_ctx.$slots, "default", { position: _ctx.position })
      ], 512);
    };
  }
});
exports._sfc_main = _sfc_main;
