// vue3-openlayers/dist/cjs/OlProfileControl.vue_vue_type_script_setup_true_lang-C1_qVhnG.js
"use strict";
const vue = require("vue");
const Profile = require("ol-ext/control/Profile");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlProfileControl",
  props: {
    className: {},
    title: {},
    style: {},
    selectStyle: {},
    info: {},
    width: {},
    height: {},
    units: {},
    feature: {},
    selectable: { type: Boolean },
    zoomable: { type: Boolean },
    numberFormat: {},
    skipFirst: {},
    element: {},
    render: { type: Function },
    target: {}
  },
  emits: ["change", "error", "propertychange", "over", "out", "show", "dragstart", "dragging", "dragend", "dragcancel"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Profile, properties, attrs);
    composables_useOpenLayersEvents.useOpenLayersEvents(control, [
      "over",
      "out",
      "show",
      "dragstart",
      "dragging",
      "dragend",
      "dragcancel"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
