// vue3-openlayers/dist/cjs/OlRotateControl.vue_vue_type_script_setup_true_lang-pW0Zajly.js
"use strict";
const vue = require("vue");
const Rotate = require("ol/control/Rotate");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlRotateControl",
  props: {
    className: {},
    label: {},
    tipLabel: {},
    compassClassName: {},
    duration: {},
    autoHide: { type: Boolean, default: true },
    render: {},
    resetNorth: {},
    target: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Rotate, properties, attrs);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
