// vue3-openlayers/dist/cjs/OlScaleLineControl.vue_vue_type_script_setup_true_lang-DtTB0LvD.js
"use strict";
const vue = require("vue");
const ScaleLine = require("ol/control/ScaleLine");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlScaleLineControl",
  props: {
    className: {},
    minWidth: {},
    maxWidth: {},
    render: { type: Function },
    target: {},
    units: {},
    bar: { type: Boolean },
    steps: {},
    text: { type: Boolean },
    dpi: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(ScaleLine, properties, attrs);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
