// vue3-openlayers/dist/cjs/OlSearchControl.vue_vue_type_script_setup_true_lang-BxXEhFPE.js
"use strict";
const vue = require("vue");
const Search = require("ol-ext/control/Search");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const composables_useOpenLayersEvents = require("./composables/useOpenLayersEvents.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSearchControl",
  props: {
    className: {},
    title: {},
    label: {},
    reverseTitle: {},
    placeholder: {},
    reverse: { type: Boolean },
    inputLabel: {},
    collapsed: { type: Boolean, default: true },
    noCollapse: { type: Boolean },
    typing: {},
    minLength: {},
    maxItems: {},
    maxHistory: {},
    getTitle: {},
    autocomplete: {},
    onselect: {},
    centerOnSelect: { type: Boolean },
    zoomOnSelect: { type: [Number, Boolean] },
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Search, properties, attrs);
    composables_useOpenLayersEvents.useOpenLayersEvents(control, ["select"]);
    __expose({ control });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
