// vue3-openlayers/dist/cjs/OlSourceCluster.vue_vue_type_script_setup_true_lang-qvcKkFwN.js
"use strict";
const vue = require("vue");
const Cluster = require("ol/source/Cluster");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceCluster",
  props: {
    attributions: {},
    distance: {},
    minDistance: {},
    geometryFunction: {},
    createCluster: {},
    source: {},
    wrapX: { type: Boolean, default: true }
  },
  emits: ["addfeature", "changefeature", "clear", "featuresloadend", "featuresloaderror", "featuresloadstart", "removefeature", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("vectorLayer");
    const { source } = composables_useSource.default(Cluster, layer, props);
    vue.provide("vectorLayer", source);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
