// vue3-openlayers/dist/cjs/OlSourceTileArcGISRest.vue_vue_type_script_setup_true_lang-s3sG9jJI.js
"use strict";
const vue = require("vue");
const TileArcGISRest = require("ol/source/TileArcGISRest");
const tilegrid = require("ol/tilegrid");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTileArcGISRest",
  props: {
    attributions: {},
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    params: {},
    hidpi: { type: Boolean, default: true },
    tileGrid: {},
    projection: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    urls: {},
    zDirection: {},
    tileSize: { default: () => [256, 256] }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const tileLayer = vue.inject("tileLayer");
    const tileGrid = vue.computed(() => {
      return props.tileGrid || tilegrid.createXYZ({
        tileSize: props.tileSize
      });
    });
    const { source } = composables_useSource.default(TileArcGISRest, tileLayer, {
      ...props,
      tileGrid: tileGrid.value
    });
    __expose({
      tileLayer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
