// vue3-openlayers/dist/cjs/OlSourceTileJSON.vue_vue_type_script_setup_true_lang-AMTOpXzn.js
"use strict";
const vue = require("vue");
const TileJSON = require("ol/source/TileJSON");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTileJSON",
  props: {
    attributions: {},
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    jsonp: { type: Boolean },
    reprojectionErrorThreshold: {},
    tileJSON: {},
    tileLoadFunction: {},
    tileSize: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(TileJSON, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
