// vue3-openlayers/dist/cjs/OlSourceBingMaps.vue_vue_type_script_setup_true_lang-DBgbszWT.js
"use strict";
const vue = require("vue");
const BingMaps = require("ol/source/BingMaps");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceBingMaps",
  props: {
    cacheSize: {},
    hidpi: { type: Boolean },
    culture: {},
    imagerySet: {},
    interpolate: { type: Boolean, default: true },
    maxZoom: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {},
    placeholderTiles: { type: Boolean },
    apiKey: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(BingMaps, layer, {
      ...props,
      key: props.apiKey
    });
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
