// vue3-openlayers/dist/cjs/OlSourceXYZ.vue_vue_type_script_setup_true_lang-Djod2An9.js
"use strict";
const vue = require("vue");
const XYZ = require("ol/source/XYZ");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceXYZ",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    projection: {},
    reprojectionErrorThreshold: {},
    maxZoom: {},
    minZoom: {},
    maxResolution: {},
    tileGrid: {},
    tileLoadFunction: {},
    tilePixelRatio: {},
    tileSize: {},
    gutter: {},
    tileUrlFunction: {},
    url: {},
    urls: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source, updateSource } = composables_useSource.default(XYZ, layer, props, [], (source2) => {
      if (props.url) {
        source2.setUrl(props.url);
      }
    });
    vue.watch(
      () => props.url,
      () => updateSource()
    );
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
