// vue3-openlayers/dist/cjs/OlSourceImageWMS.vue_vue_type_script_setup_true_lang-C_5p7dHE.js
"use strict";
const vue = require("vue");
const ImageWMS = require("ol/source/ImageWMS");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceImageWMS",
  props: {
    attributions: {},
    crossOrigin: {},
    hidpi: { type: Boolean, default: true },
    serverType: { default: "mapserver" },
    imageLoadFunction: {},
    interpolate: { type: Boolean, default: true },
    params: {},
    projection: {},
    ratio: { default: 1 },
    resolutions: {},
    url: {},
    layers: {},
    styles: { default: "" },
    time: {}
  },
  emits: ["imageloadend", "imageloaderror", "imageloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("imageLayer");
    const { source } = composables_useSource.default(ImageWMS, layer, {
      ...props,
      params: {
        ...props.params,
        LAYERS: props.layers,
        STYLES: props.styles,
        TIME: props.time
      }
    });
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
