// vue3-openlayers/dist/cjs/OlSourceGeoTIFF.vue_vue_type_script_setup_true_lang-DIvAPC3f.js
"use strict";
const vue = require("vue");
const GeoTIFF = require("ol/source/GeoTIFF");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceGeoTIFF",
  props: {
    sources: {},
    sourceOptions: {},
    convertToRGB: { type: [Boolean, String] },
    normalize: { type: Boolean, default: true },
    projection: {},
    transition: {},
    wrapX: { type: Boolean },
    interpolate: { type: Boolean, default: true }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(GeoTIFF, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
