// vue3-openlayers/dist/cjs/OlSourceTileWMS.vue_vue_type_script_setup_true_lang-P0bGA0kx.js
"use strict";
const vue = require("vue");
const TileWMS = require("ol/source/TileWMS");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTileWMS",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    params: {},
    gutter: {},
    hidpi: { type: Boolean, default: true },
    projection: { default: "EPSG:3857" },
    reprojectionErrorThreshold: {},
    tileClass: {},
    tileGrid: {},
    serverType: {},
    tileLoadFunction: {},
    url: {},
    urls: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {},
    layers: {},
    styles: { default: "" }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(TileWMS, layer, {
      ...props,
      params: {
        ...props.params,
        LAYERS: props.layers,
        STYLES: props.styles
      },
      tileGrid: props.tileGrid
    });
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
