// vue3-openlayers/dist/cjs/OlSourceVectorTile.vue_vue_type_script_setup_true_lang-CkOr3mgN.js
"use strict";
const vue = require("vue");
const VectorTileSource = require("ol/source/VectorTile");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceVectorTile",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    extent: {},
    format: {},
    overlaps: { type: Boolean, default: true },
    projection: {},
    state: {},
    tileClass: {},
    maxZoom: {},
    minZoom: {},
    tileSize: {},
    maxResolution: {},
    tileGrid: {},
    tileLoadFunction: {},
    tileUrlFunction: {},
    url: {},
    transition: {},
    urls: {},
    wrapX: { type: Boolean, default: true },
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const vectorTileLayer = vue.inject(
      "vectorTileLayer",
      null
    );
    const { source } = composables_useSource.default(VectorTileSource, vectorTileLayer, props);
    vue.provide("vectorSource", source);
    __expose({
      vectorTileLayer,
      source
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
