// vue3-openlayers/dist/cjs/OlSourceOSM.vue_vue_type_script_setup_true_lang-NXeE6VQ-.js
"use strict";
const vue = require("vue");
const OSM = require("ol/source/OSM");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceOSM",
  props: {
    attributions: {},
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    maxZoom: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    transition: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    zDirection: {}
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(OSM, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
