// vue3-openlayers/dist/cjs/OlSourceWMTS.vue_vue_type_script_setup_true_lang-CktmXUoe.js
"use strict";
const vue = require("vue");
const WMTSSource = require("ol/source/WMTS");
const TileGridWMTS = require("ol/tilegrid/WMTS");
const proj = require("ol/proj");
const extent = require("ol/extent");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceWMTS",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean, default: true },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean, default: true },
    projection: {},
    reprojectionErrorThreshold: {},
    requestEncoding: {},
    layer: {},
    tileClass: {},
    tilePixelRatio: {},
    format: {},
    version: {},
    matrixSet: {},
    dimensions: {},
    url: {},
    tileLoadFunction: {},
    urls: {},
    wrapX: { type: Boolean },
    transition: {},
    zDirection: {},
    tileGrid: {},
    styles: {},
    tileZoomLevel: { default: 30 },
    tileMatrixPrefix: { default: "" }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const tileLayer = vue.inject("tileLayer");
    const extent$1 = vue.computed(() => {
      return proj.get(props.projection)?.getExtent();
    });
    const origin = vue.computed(() => {
      return extent$1.value ? extent.getTopLeft(extent$1.value) : void 0;
    });
    const size = vue.computed(() => {
      return extent$1.value ? extent.getWidth(extent$1.value) / 256 : 0;
    });
    const tileGrid = vue.computed(() => {
      if (props.tileGrid) {
        return props.tileGrid;
      }
      const resolutions = [props.tileZoomLevel];
      const matrixIds = [`${props.tileZoomLevel}`];
      for (let z = 0; z < props.tileZoomLevel; ++z) {
        resolutions[z] = size.value / Math.pow(2, z);
        matrixIds[z] = props.tileMatrixPrefix + z;
      }
      return new TileGridWMTS({
        origin: origin.value,
        resolutions,
        matrixIds
      });
    });
    const { source } = composables_useSource.default(WMTSSource, tileLayer, {
      ...props,
      style: props.styles,
      tileGrid: tileGrid.value
    });
    __expose({
      tileLayer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
