// vue3-openlayers/dist/cjs/OlSourceVector.vue_vue_type_script_setup_true_lang-Bmh4VJNI.js
"use strict";
const vue = require("vue");
const VectorSource = require("ol/source/Vector");
require("ol/layer/WebGLVector");
const composables_useSource = require("./composables/useSource.js");
const ol = require("ol");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceVector",
  props: {
    attributions: {},
    features: {},
    format: {},
    loader: {},
    overlaps: { type: Boolean, default: true },
    strategy: {},
    url: {},
    useSpatialIndex: { type: Boolean, default: true },
    wrapX: { type: Boolean, default: true }
  },
  emits: ["addfeature", "changefeature", "clear", "featuresloadend", "featuresloaderror", "featuresloadstart", "removefeature", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const vectorLayer = vue.inject(
      "vectorLayer",
      null
    );
    const heatmapLayer = vue.inject("heatmapLayer", null);
    const webglVectorLayer = vue.inject(
      "webglVectorLayer",
      null
    );
    const layer = heatmapLayer || vectorLayer || webglVectorLayer;
    const { source, updateSource } = composables_useSource.default(VectorSource, layer, props);
    vue.watch(
      () => props.features,
      (updatedFeatures, oldFeatures) => {
        if (updatedFeatures !== oldFeatures) {
          source.value.clear();
          if (updatedFeatures) {
            const featureArray = updatedFeatures instanceof ol.Collection ? updatedFeatures.getArray() : updatedFeatures;
            source.value.addFeatures(featureArray);
          }
        }
        updateSource();
      },
      { deep: true }
    );
    vue.watch(
      () => props.url,
      (updatedUrl, newUrl) => {
        if (updatedUrl !== newUrl) {
          source.value.clear();
          if (updatedUrl) {
            source.value.setUrl(updatedUrl);
            source.value.refresh();
          }
        }
        updateSource();
      }
    );
    vue.provide("vectorSource", source);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
