// vue3-openlayers/dist/cjs/OlSourceStadiaMaps.vue_vue_type_script_setup_true_lang-CZC_A0yg.js
"use strict";
const vue = require("vue");
const StadiaMaps = require("ol/source/StadiaMaps");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceStadiaMaps",
  props: {
    cacheSize: {},
    interpolate: { type: Boolean, default: true },
    layer: {},
    minZoom: {},
    maxZoom: {},
    reprojectionErrorThreshold: {},
    tileLoadFunction: {},
    transition: {},
    url: {},
    wrapX: { type: Boolean, default: true },
    zDirection: {},
    apiKey: {},
    retina: { type: Boolean }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(StadiaMaps, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
