// vue3-openlayers/dist/cjs/OlSourceTianditu.vue_vue_type_script_setup_true_lang-Knj7uFl5.js
"use strict";
const vue = require("vue");
const sources_TiandituClass = require("./sources/TiandituClass.js");
const composables_useSource = require("./composables/useSource.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlSourceTianditu",
  props: {
    attributions: {},
    attributionsCollapsible: { type: Boolean },
    cacheSize: {},
    crossOrigin: {},
    interpolate: { type: Boolean },
    tileGrid: {},
    projection: { default: "EPSG:3857" },
    reprojectionErrorThreshold: {},
    requestEncoding: { default: "KVP" },
    layer: {},
    style: {},
    tileClass: {},
    tilePixelRatio: {},
    format: {},
    version: { default: "1.0.0" },
    matrixSet: {},
    dimensions: { default: () => ({}) },
    url: {},
    tileLoadFunction: { type: Function, default: (imageTile, src) => {
      imageTile.getImage().src = src;
    } },
    urls: {},
    wrapX: { type: Boolean, default: true },
    transition: {},
    zDirection: {},
    tileProxy: {},
    layerType: { default: "img" },
    tk: {},
    isLabel: { type: Boolean, default: false },
    maxZoom: { default: 21 }
  },
  emits: ["tileloadend", "tileloaderror", "tileloadstart", "change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const layer = vue.inject("tileLayer");
    const { source } = composables_useSource.default(sources_TiandituClass.default, layer, props);
    __expose({
      layer,
      source
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
