// vue3-openlayers/dist/cjs/OlStyleText.vue_vue_type_script_setup_true_lang-DdQ9mKeQ.js
"use strict";
const vue = require("vue");
const Text = require("ol/style/Text");
const Fill = require("ol/style/Fill");
const Stroke = require("ol/style/Stroke");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const interaction = require("ol/interaction");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlStyleText",
  props: {
    font: {},
    maxAngle: {},
    offsetX: {},
    offsetY: {},
    overflow: { type: Boolean },
    placement: {},
    repeat: {},
    scale: {},
    rotateWithView: { type: Boolean },
    keepUpright: { type: Boolean },
    rotation: {},
    text: {},
    textAlign: {},
    justify: {},
    textBaseline: {},
    fill: {},
    stroke: {},
    backgroundFill: {},
    backgroundStroke: {},
    padding: {},
    declutterMode: {}
  },
  setup(__props) {
    const props = __props;
    const style = vue.inject("style", null);
    const styledObj = vue.inject(
      "styledObj",
      null
    );
    const properties = composables_usePropsAsObjectProperties.default(props);
    const createText = (properties2) => {
      const innerProperties = properties2;
      const options = {
        ...innerProperties
      };
      if (properties2.fill) {
        options.fill = new Fill({ color: properties2.fill });
      }
      if (properties2.stroke) {
        options.stroke = new Stroke(properties2.stroke);
      }
      if (properties2.backgroundFill) {
        options.backgroundFill = new Fill({ color: properties2.backgroundFill });
      }
      if (properties2.backgroundStroke) {
        options.backgroundStroke = new Stroke(properties2.backgroundStroke);
      }
      return new Text(options);
    };
    const textContent = vue.computed(() => createText(properties));
    const applyStyle = () => {
      style?.value?.setText(new Text());
      style?.value?.setText(textContent.value);
      if (styledObj?.value instanceof interaction.Interaction) {
        styledObj.value.changed();
      }
    };
    vue.watch(properties, () => applyStyle());
    vue.watch(
      () => style?.value,
      () => applyStyle()
    );
    vue.onMounted(() => style?.value?.setText(textContent.value));
    vue.onUnmounted(() => style?.value?.setText(new Text()));
    vue.provide("style", textContent);
    vue.provide("styledObj", styledObj);
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
