// vue3-openlayers/dist/cjs/OlStyleFlowLine.vue_vue_type_script_setup_true_lang-I4sbfl4Q.js
"use strict";
const vue = require("vue");
const FlowLine = require("ol-ext/style/FlowLine");
const Draw = require("ol/interaction/Draw");
const Modify = require("ol/interaction/Modify");
const Feature = require("ol/Feature");
const Style = require("ol/style/Style");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
require("ol/interaction");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlStyleFlowLine",
  props: {
    visible: { type: Boolean },
    width: {},
    width2: {},
    color: {},
    color2: {},
    arrowClor: {},
    lineCap: { default: "butt" },
    arrowSize: {},
    arrow: {},
    noOverlap: { type: Boolean },
    offset0: {},
    offset1: {},
    overrideStyleFunction: {}
  },
  setup(__props, { expose: __expose }) {
    const props = __props;
    const styledObj = vue.inject(
      "stylable",
      null
    );
    const properties = composables_usePropsAsObjectProperties.default(props);
    const style = vue.computed(() => new FlowLine(properties));
    const setStyle = (val) => {
      if (!styledObj?.value) {
        return;
      }
      if (styledObj.value instanceof Draw || styledObj?.value instanceof Modify) {
        styledObj.value.getOverlay().setStyle(val);
        styledObj.value.dispatchEvent("styleChanged");
        return;
      }
      try {
        if (styledObj.value instanceof Feature) {
          styledObj.value.setStyle(val);
        }
        styledObj.value.changed();
        styledObj.value.dispatchEvent("styleChanged");
      } catch (error) {
        if (styledObj?.value) {
          styledObj.value.style_ = val;
          styledObj.value.values_.style = val;
          styledObj.value.changed();
          styledObj.value.dispatchEvent("styleChanged");
        }
      }
    };
    const styleFunc = vue.computed(() => {
      return (feature, resolution) => {
        if (properties.overrideStyleFunction) {
          properties.overrideStyleFunction(feature, style.value, resolution);
        }
        return style.value;
      };
    });
    vue.watch(
      properties,
      () => setStyle(properties.overrideStyleFunction ? styleFunc.value : style.value)
    );
    vue.onMounted(
      () => setStyle(properties.overrideStyleFunction ? styleFunc.value : style.value)
    );
    vue.onUnmounted(() => setStyle(new Style()));
    __expose({
      style
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
