// vue3-openlayers/dist/cjs/OlStyleStroke.vue_vue_type_script_setup_true_lang-T9R5s0Xw.js
"use strict";
const vue = require("vue");
const Stroke = require("ol/style/Stroke");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const interaction = require("ol/interaction");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlStyleStroke",
  props: {
    color: {},
    lineCap: {},
    lineJoin: {},
    lineDash: {},
    lineDashOffset: {},
    miterLimit: {},
    width: { default: 1 }
  },
  setup(__props) {
    const props = __props;
    const style = vue.inject("style", null);
    const styledObj = vue.inject(
      "styledObj",
      null
    );
    const circle = vue.inject("circle", null);
    const properties = composables_usePropsAsObjectProperties.default(props);
    if (style != null && circle == null) {
      let stroke = new Stroke(properties);
      style.value?.setStroke(stroke);
      const applyStroke = () => {
        style?.value?.setStroke(new Stroke());
        stroke = new Stroke(properties);
        style?.value?.setStroke(stroke);
      };
      vue.watch(properties, () => {
        applyStroke();
      });
      vue.watch(style, () => {
        applyStroke();
      });
      vue.onMounted(() => {
        style?.value?.setStroke(stroke);
      });
      vue.onUnmounted(() => {
        style?.value?.setStroke(new Stroke());
      });
    } else if (circle != null) {
      const applyStroketoCircle = (innerProperties) => {
        circle?.value?.getStroke()?.setColor(innerProperties.color || "");
        circle?.value?.getStroke()?.setLineCap(innerProperties.lineCap);
        circle?.value?.getStroke()?.setLineDash(innerProperties.lineDash || []);
        circle?.value?.getStroke()?.setLineDashOffset(innerProperties.lineDashOffset);
        circle?.value?.getStroke()?.setLineJoin(innerProperties.lineJoin);
        circle?.value?.getStroke()?.setMiterLimit(innerProperties.miterLimit);
        circle?.value?.getStroke()?.setWidth(innerProperties.width);
        circle?.value?.setRadius(circle?.value?.getRadius());
        if (styledObj?.value instanceof interaction.Interaction) {
          styledObj.value.changed();
        }
      };
      applyStroketoCircle(properties);
      vue.watch(properties, (newVal) => {
        applyStroketoCircle(newVal);
      });
      vue.watch(circle, () => {
        applyStroketoCircle(properties);
      });
    }
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
