// vue3-openlayers/dist/cjs/OlStyleCircle.vue_vue_type_script_setup_true_lang-BkMfTzVy.js
"use strict";
const vue = require("vue");
const CircleStyle = require("ol/style/Circle");
const Fill = require("ol/style/Fill");
const Stroke = require("ol/style/Stroke");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const interaction = require("ol/interaction");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlStyleCircle",
  props: {
    radius: {},
    scale: {}
  },
  setup(__props) {
    const props = __props;
    const style = vue.inject("style", null);
    const styledObj = vue.inject(
      "styledObj",
      null
    );
    const properties = composables_usePropsAsObjectProperties.default(props);
    const createCircleStyle = (innerProperties) => {
      return new CircleStyle({
        ...innerProperties,
        fill: new Fill(),
        stroke: new Stroke()
      });
    };
    const circle = vue.computed(() => createCircleStyle(properties));
    const applyStyle = () => {
      style?.value?.setImage(circle.value);
      if (styledObj?.value instanceof interaction.Interaction) {
        styledObj.value.changed();
      }
    };
    vue.watch(properties, () => applyStyle());
    vue.watch(
      () => style,
      () => applyStyle()
    );
    vue.onMounted(() => applyStyle());
    vue.onUnmounted(() => {
      style?.value?.setImage(null);
    });
    vue.provide("circle", circle);
    vue.provide("styledObj", styledObj);
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
