// vue3-openlayers/dist/cjs/OlSwipeControl.vue_vue_type_script_setup_true_lang-i1JfGijn.js
"use strict";
const vue = require("vue");
const Swipe = require("ol-ext/control/Swipe");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlSwipeControl",
  props: {
    layers: {},
    rightLayers: {},
    className: { default: "ol-swipe" },
    position: {},
    orientation: {},
    element: {},
    render: {},
    target: {}
  },
  emits: ["change", "error", "propertychange", "moving"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Swipe, properties, attrs, ["moving"]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
