// vue3-openlayers/dist/cjs/OlTileLayer.vue_vue_type_script_setup_true_lang-BInKQGek.js
"use strict";
const vue = require("vue");
const TileLayer = require("ol/layer/Tile");
const composables_useLayer = require("./composables/useLayer.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
require("ol/source/Tile");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlTileLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    preload: {},
    source: {},
    map: {},
    background: { type: [String, Function] },
    useInterimTilesOnError: { type: Boolean },
    properties: {},
    cacheSize: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, layers_LayersCommonProps.useDefaults({
    preload: 1
  })),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender", "change:useInterimTilesOnError", "change:preload"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = composables_useLayer.default(TileLayer, vue.toRefs(props), [
      "change:useInterimTilesOnError",
      "change:preload"
    ]);
    vue.provide("tileLayer", layer);
    __expose({
      tileLayer: layer
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
