// vue3-openlayers/dist/cjs/OlToggleControl.vue_vue_type_script_setup_true_lang-CVyzXOju.js
"use strict";
const vue = require("vue");
const Toggle = require("ol-ext/control/Toggle");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlToggleControl",
  props: {
    className: {},
    classButton: {},
    title: {},
    html: {},
    interaction: {},
    active: { type: Boolean },
    disable: { type: Boolean },
    bar: {},
    autoActive: { type: Boolean },
    onToggle: { type: Function },
    attributes: {},
    element: {},
    render: { type: Function },
    target: {}
  },
  emits: ["change", "error", "propertychange", "change:active", "change:disable"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Toggle, properties, attrs, [
      "change:active",
      "change:disable"
    ]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
