// vue3-openlayers/dist/cjs/OlVectorTileLayer.vue_vue_type_script_setup_true_lang-Cc_S7a6J.js
"use strict";
const vue = require("vue");
const VectorTileLayer = require("ol/layer/VectorTile");
const composables_useLayer = require("./composables/useLayer.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlVectorTileLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    renderOrder: { type: Function },
    renderBuffer: {},
    renderMode: {},
    source: {},
    map: {},
    declutter: { type: [String, Number, Boolean] },
    style: { type: [Object, Array, Function, null] },
    background: { type: [String, Function] },
    updateWhileAnimating: { type: Boolean },
    updateWhileInteracting: { type: Boolean },
    preload: {},
    useInterimTilesOnError: { type: Boolean },
    properties: {},
    cacheSize: {},
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {}
  }, layers_LayersCommonProps.useDefaults()),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = composables_useLayer.default(VectorTileLayer, props);
    vue.provide("vectorTileLayer", layer);
    vue.provide("stylable", layer);
    __expose({
      vectorTileLayer: layer
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
