// vue3-openlayers/dist/cjs/OlVideoRecorderControl.vue_vue_type_script_setup_true_lang-CBfl6-Nq.js
"use strict";
const vue = require("vue");
const VideoRecorder = require("ol-ext/control/VideoRecorder");
const FileSaver_min = require("./FileSaver.min-B_I2ImpY.js");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  ...{
    inheritAttrs: false
  },
  __name: "OlVideoRecorderControl",
  props: {
    className: {},
    frameRate: {},
    videoBitsPerSecond: {},
    videoTarget: {},
    element: {},
    render: {},
    target: {},
    downloadName: { default: "mapVideo.mp4" }
  },
  emits: ["change", "error", "propertychange", "start", "stop", "pause", "result"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(VideoRecorder, properties, attrs, [
      "start",
      "stop",
      "pause",
      "resume"
    ]);
    control.value.on(
      // @ts-expect-error because of wrong typings
      "stop",
      (event) => {
        if (props.downloadName) {
          FileSaver_min.FileSaver_minExports.saveAs(event.videoURL, props.downloadName);
        }
      }
    );
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
