// vue3-openlayers/dist/cjs/OlWebglVectorLayer.vue_vue_type_script_setup_true_lang-Cry9xLh6.js
"use strict";
const vue = require("vue");
const composables_useLayer = require("./composables/useLayer.js");
const layers_LayersCommonProps = require("./layers/LayersCommonProps.js");
const WebGLVectorLayer = require("ol/layer/WebGLVector");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlWebglVectorLayer",
  props: /* @__PURE__ */ vue.mergeDefaults({
    className: {},
    opacity: {},
    visible: { type: Boolean },
    extent: {},
    zIndex: {},
    minResolution: {},
    maxResolution: {},
    minZoom: {},
    maxZoom: {},
    source: {},
    variables: {},
    background: { type: [String, Function] },
    disableHitDetection: { type: Boolean },
    properties: {},
    map: {},
    render: { type: Function },
    openInLayerSwitcher: { type: Boolean },
    title: {},
    name: {},
    allwaysOnTop: { type: Boolean },
    baseLayer: { type: Boolean },
    displayInLayerSwitcher: { type: Boolean },
    preview: {},
    styles: {}
  }, layers_LayersCommonProps.useDefaults({
    styles: () => ({
      "shape-points": 1,
      "shape-radius": 10,
      "shape-opacity": 0.5,
      "shape-fill-color": "blue"
    })
  })),
  emits: ["sourceready", "change", "error", "propertychange", "change:extent", "change:maxResolution", "change:maxZoom", "change:minResolution", "change:minZoom", "change:opacity", "change:source", "change:visible", "change:zIndex", "postrender", "prerender"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const { layer } = composables_useLayer.default(WebGLVectorLayer, props);
    vue.provide("webglVectorLayer", layer);
    vue.watch(
      () => props.variables,
      (newVariables) => {
        if (newVariables) {
          layer.value.updateStyleVariables(newVariables);
        }
      }
    );
    __expose({
      webglVectorLayer: layer
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.renderSlot(_ctx.$slots, "default")
      ]);
    };
  }
});
exports._sfc_main = _sfc_main;
