// vue3-openlayers/dist/cjs/OlZoneControl.vue_vue_type_script_setup_true_lang-ByZPW62w.js
"use strict";
const vue = require("vue");
const MapZone = require("ol-ext/control/MapZone");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlZoneControl",
  props: {
    className: { default: "ol-mapzone" },
    layer: {},
    projection: { default: "EPSG:3857" },
    centerOnClick: { type: Boolean, default: true },
    element: {},
    render: {},
    target: {},
    zones: {}
  },
  emits: ["change", "error", "propertychange", "select"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(MapZone, properties, attrs, ["select"]);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
