// vue3-openlayers/dist/cjs/OlZoomControl.vue_vue_type_script_setup_true_lang-DJWyxvPx.js
"use strict";
const vue = require("vue");
const Zoom = require("ol/control/Zoom");
const composables_useControl = require("./composables/useControl.js");
const composables_usePropsAsObjectProperties = require("./composables/usePropsAsObjectProperties.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "OlZoomControl",
  props: {
    duration: {},
    className: {},
    zoomInClassName: {},
    zoomOutClassName: {},
    zoomInLabel: {},
    zoomOutLabel: {},
    zoomInTipLabel: {},
    zoomOutTipLabel: {},
    delta: {},
    target: {}
  },
  emits: ["change", "error", "propertychange"],
  setup(__props, { expose: __expose }) {
    const props = __props;
    const attrs = vue.useAttrs();
    const properties = composables_usePropsAsObjectProperties.default(props);
    const { control } = composables_useControl.default(Zoom, properties, attrs);
    __expose({
      control
    });
    return (_ctx, _cache) => {
      return vue.createCommentVNode("", true);
    };
  }
});
exports._sfc_main = _sfc_main;
