// vue3-openlayers/dist/esm/styles/index.js
import { _ as _sfc_main } from "../OlStyle.vue_vue_type_script_setup_true_lang-DccAE5_b.js";
import { _ as _sfc_main$1 } from "../OlStyleCircle.vue_vue_type_script_setup_true_lang-J9mmiSKo.js";
import { _ as _sfc_main$2 } from "../OlStyleStroke.vue_vue_type_script_setup_true_lang-DZedA3zE.js";
import { _ as _sfc_main$3 } from "../OlStyleFill.vue_vue_type_script_setup_true_lang-BzTzeJPJ.js";
import { _ as _sfc_main$4 } from "../OlStyleIcon.vue_vue_type_script_setup_true_lang-DAFvao6G.js";
import { _ as _sfc_main$5 } from "../OlStyleText.vue_vue_type_script_setup_true_lang-qgBomPf-.js";
import { _ as _sfc_main$6 } from "../OlStyleFlowLine.vue_vue_type_script_setup_true_lang-DBMVv7NG.js";
import { r as registerWithAliases } from "../registerWithAliases-Jl4Xktq1.js";
function install(app, options) {
  registerWithAliases(app, "OlStyle", _sfc_main);
  registerWithAliases(app, "OlStyleCircle", _sfc_main$1);
  registerWithAliases(app, "OlStyleStroke", _sfc_main$2);
  registerWithAliases(app, "OlStyleFill", _sfc_main$3);
  registerWithAliases(app, "OlStyleIcon", _sfc_main$4);
  registerWithAliases(app, "OlStyleText", _sfc_main$5);
  registerWithAliases(app, "OlStyleFlowLine", _sfc_main$6);
  if (options) {
    app.provide("ol-options", options);
  }
}
export {
  _sfc_main as OlStyle,
  _sfc_main$1 as OlStyleCircle,
  _sfc_main$3 as OlStyleFill,
  _sfc_main$6 as OlStyleFlowLine,
  _sfc_main$4 as OlStyleIcon,
  _sfc_main$2 as OlStyleStroke,
  _sfc_main$5 as OlStyleText,
  install as default,
  install
};
